// @aislamov/onnxruntime-web64/lib/wasm/binding/ort-wasm-simd.jsep-copy5-working.js

var ortWasm = (() => {
  var _scriptDir = typeof document !== 'undefined' && document.currentScript ? document.currentScript.src : undefined;
  if (typeof __filename !== 'undefined') _scriptDir = _scriptDir || __filename;
  return (
function(ortWasm = {})  {

var Module = typeof ortWasm != "undefined" ? ortWasm : {};

var readyPromiseResolve, readyPromiseReject;

Module["ready"] = new Promise((resolve, reject) => {
 readyPromiseResolve = resolve;
 readyPromiseReject = reject;
});

var moduleOverrides = Object.assign({}, Module);

var arguments_ = [];

var thisProgram = "./this.program";

var quit_ = (status, toThrow) => {
 throw toThrow;
};

var ENVIRONMENT_IS_WEB = typeof window == "object";

var ENVIRONMENT_IS_WORKER = typeof importScripts == "function";

var ENVIRONMENT_IS_NODE = typeof process == "object" && typeof process.versions == "object" && typeof process.versions.node == "string";

var ENVIRONMENT_IS_SHELL = !ENVIRONMENT_IS_WEB && !ENVIRONMENT_IS_NODE && !ENVIRONMENT_IS_WORKER;

var scriptDirectory = "";

function locateFile(path) {
 if (Module["locateFile"]) {
  return Module["locateFile"](path, scriptDirectory);
 }
 return scriptDirectory + path;
}

var read_, readAsync, readBinary, setWindowTitle;

if (ENVIRONMENT_IS_NODE) {
 var fs = require("fs");
 var nodePath = require("path");
 if (ENVIRONMENT_IS_WORKER) {
  scriptDirectory = nodePath.dirname(scriptDirectory) + "/";
 } else {
  scriptDirectory = __dirname + "/";
 }
 read_ = (filename, binary) => {
  filename = isFileURI(filename) ? new URL(filename) : nodePath.normalize(filename);
  return fs.readFileSync(filename, binary ? undefined : "utf8");
 };
 readBinary = filename => {
  var ret = read_(filename, true);
  if (!ret.buffer) {
   ret = new Uint8Array(ret);
  }
  return ret;
 };
 readAsync = (filename, onload, onerror, binary = true) => {
  filename = isFileURI(filename) ? new URL(filename) : nodePath.normalize(filename);
  fs.readFile(filename, binary ? undefined : "utf8", (err, data) => {
   if (err) onerror(err); else onload(binary ? data.buffer : data);
  });
 };
 if (!Module["thisProgram"] && process.argv.length > 1) {
  thisProgram = process.argv[1].replace(/\\/g, "/");
 }
 arguments_ = process.argv.slice(2);
 quit_ = (status, toThrow) => {
  process.exitCode = status;
  throw toThrow;
 };
 Module["inspect"] = () => "[Emscripten Module object]";
} else if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
 if (ENVIRONMENT_IS_WORKER) {
  scriptDirectory = self.location.href;
 } else if (typeof document != "undefined" && document.currentScript) {
  scriptDirectory = document.currentScript.src;
 }
 if (_scriptDir) {
  scriptDirectory = _scriptDir;
 }
 if (scriptDirectory.indexOf("blob:") !== 0) {
  scriptDirectory = scriptDirectory.substr(0, scriptDirectory.replace(/[?#].*/, "").lastIndexOf("/") + 1);
 } else {
  scriptDirectory = "";
 }
 {
  read_ = url => {
   var xhr = new XMLHttpRequest();
   xhr.open("GET", url, false);
   xhr.send(null);
   return xhr.responseText;
  };
  if (ENVIRONMENT_IS_WORKER) {
   readBinary = url => {
    var xhr = new XMLHttpRequest();
    xhr.open("GET", url, false);
    xhr.responseType = "arraybuffer";
    xhr.send(null);
    return new Uint8Array(xhr.response);
   };
  }
  readAsync = (url, onload, onerror) => {
   var xhr = new XMLHttpRequest();
   xhr.open("GET", url, true);
   xhr.responseType = "arraybuffer";
   xhr.onload = () => {
    if (xhr.status == 200 || xhr.status == 0 && xhr.response) {
     onload(xhr.response);
     return;
    }
    onerror();
   };
   xhr.onerror = onerror;
   xhr.send(null);
  };
 }
 setWindowTitle = title => document.title = title;
} else {}

var out = Module["print"] || console.log.bind(console);

var err = Module["printErr"] || console.error.bind(console);

Object.assign(Module, moduleOverrides);

moduleOverrides = null;

if (Module["arguments"]) arguments_ = Module["arguments"];

if (Module["thisProgram"]) thisProgram = Module["thisProgram"];

if (Module["quit"]) quit_ = Module["quit"];

var wasmBinary;

if (Module["wasmBinary"]) wasmBinary = Module["wasmBinary"];

var noExitRuntime = Module["noExitRuntime"] || true;

if (typeof WebAssembly != "object") {
 abort("no native wasm support detected");
}

var wasmMemory;

var ABORT = false;

var EXITSTATUS;

function assert(condition, text) {
 if (!condition) {
  abort(text);
 }
}

var HEAP, HEAP8, HEAPU8, HEAP16, HEAPU16, HEAP32, HEAPU32, HEAPF32, HEAP64, HEAPU64, HEAPF64;

const getHeapBlock = (type, offset, byteLength) => {
 const maxMemory = 4 * 1024 * 1024 * 1e3;
 const maxSize = Math.min(wasmMemory.buffer.byteLength, maxMemory);
 const heap = wasmMemory.buffer;
 switch (type) {
 case "i1":
 case "i8":
  return new Int8Array(heap, offset, byteLength || maxSize / Int8Array.BYTES_PER_ELEMENT);

 case "u1":
 case "u8":
  return new Uint8Array(heap, offset, byteLength || maxSize / Uint8Array.BYTES_PER_ELEMENT);

 case "i16":
  return new Int16Array(heap, offset, byteLength || maxSize / Int16Array.BYTES_PER_ELEMENT);

 case "u16":
  return new Uint16Array(heap, offset, byteLength || maxSize / Uint16Array.BYTES_PER_ELEMENT);

 case "i32":
  return new Int32Array(heap, offset, byteLength || maxSize / Int32Array.BYTES_PER_ELEMENT);

 case "u32":
  return new Uint32Array(heap, offset, byteLength || maxSize / Uint32Array.BYTES_PER_ELEMENT);

 case "f32":
  return new Float32Array(heap, offset, byteLength || maxSize / Float32Array.BYTES_PER_ELEMENT);

 case "f64":
  return new Float64Array(heap, offset, byteLength || maxSize / Float64Array.BYTES_PER_ELEMENT);

 case "i64":
  return new BigInt64Array(heap, offset, byteLength || maxSize / BigInt64Array.BYTES_PER_ELEMENT);

 case "*":
 case "u64":
  return new BigUint64Array(heap, offset, byteLength || maxSize / BigUint64Array.BYTES_PER_ELEMENT);

 default:
  throw new Error("Invalid type");
 }
};

function updateMemoryViews() {
 var b = wasmMemory.buffer;
 var maxSize = Math.min(b.byteLength, 4 * 1024 * 1024 * 1e3);
 const maxMemory = 4 * 1024 * 1024 * 1e3;
 var proxyHandler = (type, hb) => ({
  heapBlock: hb,
  copyWithin: function(target, start, end) {
   const bpe = hb.BYTES_PER_ELEMENT;
   if (target * bpe >= maxSize || start * bpe >= maxSize || end && end * bpe >= maxSize) {
    // console.log("copy within overridden", target, start, end, maxSize);
    const len = (end - start) * bpe;
    const targetArray = getHeapBlock(type, target * bpe, len);
    const sourceArray = getHeapBlock(type, start * bpe, len);
    targetArray.set(sourceArray);
    return hb;
   } else {
    return hb.copyWithin(target, start, end);
   }
  },
  setOverridden: function(array, offset) {
   const offsetReal = (offset || 0) * hb.BYTES_PER_ELEMENT;
   if (offsetReal >= maxSize || array.byteLength + offsetReal >= maxSize) {
    // console.log("set overridden", offset);
    const targetArray = getHeapBlock(type, offsetReal, array.byteLength);
    targetArray.set(array);
   } else {
    hb.set(array, offset);
   }
  },
  get: function(target, property) {
   if (typeof property === "number" || typeof property === "bigint") {
    var memoryOffset = Number(property) * target.BYTES_PER_ELEMENT;
    if (memoryOffset >= maxMemory) {
     // console.log("get overridden", property);
     var heap = getHeapBlock(type, memoryOffset, target.BYTES_PER_ELEMENT * 2);
     return heap[0 >>> 0];
    } else {
     return this.heapBlock[property];
    }
   }
   if (property === "copyWithin") {
    return this.copyWithin;
   }
   if (property === "set") {
    return this.setOverridden;
   }
   if (property === "subarray") {
    const subarray = (start, end) => {
     const startReal = (start || 0) * this.heapBlock.BYTES_PER_ELEMENT;
     const endReal = end ? end * this.heapBlock.BYTES_PER_ELEMENT : wasmMemory.byteLength;
     if (startReal >= maxSize || endReal >= maxSize) {
      // console.log("subarray overridden", start, end);
      return getHeapBlock(type, startReal, endReal - startReal);
     } else {
      return this.heapBlock.subarray(start, end);
     }
    };
    return subarray;
   }
   console.log("returning default", property);
   return this.heapBlock[property];
  },
  set: function(target, property, value) {
   if (typeof property === "number" || typeof property === "bigint") {
    var memoryOffset = Number(property) * target.BYTES_PER_ELEMENT;
    if (memoryOffset >= maxMemory) {
     // console.log("set overridden", property);
     var heap = getHeapBlock(type, memoryOffset, target.BYTES_PER_ELEMENT);
     heap[0 >>> 0] = value;
     return true;
    }
   }
   this.heapBlock[property] = value;
   return true;
  }
 });
 console.log("max size", maxSize, b.byteLength);
 function createMemoryProxy(type) {
  const block = getHeapBlock(type, 0);
  return new Proxy(block, proxyHandler(type, block));
 }
 try {
  Module["HEAP8"] = HEAP8 = createMemoryProxy("i8");
 } catch (e) {
  console.error(e);
  throw e;
 }
 Module["HEAP16"] = HEAP16 = createMemoryProxy("i16");
 Module["HEAP32"] = HEAP32 = createMemoryProxy("i32");
 Module["HEAPU8"] = HEAPU8 = createMemoryProxy("u8");
 Module["HEAPU16"] = HEAPU16 = createMemoryProxy("u16");
 Module["HEAPU32"] = HEAPU32 = createMemoryProxy("u32");
 Module["HEAPF32"] = HEAPF32 = createMemoryProxy("f32");
 Module["HEAPF64"] = HEAPF64 = createMemoryProxy("f64");
 Module["HEAP64"] = HEAP64 = createMemoryProxy("i64");
 Module["HEAPU64"] = HEAPU64 = createMemoryProxy("u64");
 Module["MEMORY"] = b;
}

var wasmTable;

var __ATPRERUN__ = [];

var __ATINIT__ = [];

var __ATEXIT__ = [];

var __ATPOSTRUN__ = [];

var runtimeInitialized = false;

var runtimeKeepaliveCounter = 0;

function keepRuntimeAlive() {
 return noExitRuntime || runtimeKeepaliveCounter > 0;
}

function preRun() {
 if (Module["preRun"]) {
  if (typeof Module["preRun"] == "function") Module["preRun"] = [ Module["preRun"] ];
  while (Module["preRun"].length) {
   addOnPreRun(Module["preRun"].shift());
  }
 }
 callRuntimeCallbacks(__ATPRERUN__);
}

function initRuntime() {
 runtimeInitialized = true;
 callRuntimeCallbacks(__ATINIT__);
}

function postRun() {
 if (Module["postRun"]) {
  if (typeof Module["postRun"] == "function") Module["postRun"] = [ Module["postRun"] ];
  while (Module["postRun"].length) {
   addOnPostRun(Module["postRun"].shift());
  }
 }
 callRuntimeCallbacks(__ATPOSTRUN__);
}

function addOnPreRun(cb) {
 __ATPRERUN__.unshift(cb);
}

function addOnInit(cb) {
 __ATINIT__.unshift(cb);
}

function addOnExit(cb) {}

function addOnPostRun(cb) {
 __ATPOSTRUN__.unshift(cb);
}

var runDependencies = 0;

var runDependencyWatcher = null;

var dependenciesFulfilled = null;

function getUniqueRunDependency(id) {
 return id;
}

function addRunDependency(id) {
 runDependencies++;
 if (Module["monitorRunDependencies"]) {
  Module["monitorRunDependencies"](runDependencies);
 }
}

function removeRunDependency(id) {
 runDependencies--;
 if (Module["monitorRunDependencies"]) {
  Module["monitorRunDependencies"](runDependencies);
 }
 if (runDependencies == 0) {
  if (runDependencyWatcher !== null) {
   clearInterval(runDependencyWatcher);
   runDependencyWatcher = null;
  }
  if (dependenciesFulfilled) {
   var callback = dependenciesFulfilled;
   dependenciesFulfilled = null;
   callback();
  }
 }
}

function abort(what) {
 if (Module["onAbort"]) {
  Module["onAbort"](what);
 }
 what = "Aborted(" + what + ")";
 err(what);
 ABORT = true;
 EXITSTATUS = 1;
 what += ". Build with -sASSERTIONS for more info.";
 var e = new WebAssembly.RuntimeError(what);
 readyPromiseReject(e);
 throw e;
}

var dataURIPrefix = "data:application/octet-stream;base64,";

function isDataURI(filename) {
 return filename.startsWith(dataURIPrefix);
}

function isFileURI(filename) {
 return filename.startsWith("file://");
}

var wasmBinaryFile;

wasmBinaryFile = "ort-wasm-simd.wasm";

if (!isDataURI(wasmBinaryFile)) {
 wasmBinaryFile = locateFile(wasmBinaryFile);
}

function getBinary(file) {
 try {
  if (file == wasmBinaryFile && wasmBinary) {
   return new Uint8Array(wasmBinary);
  }
  if (readBinary) {
   return readBinary(file);
  }
  throw "both async and sync fetching of the wasm failed";
 } catch (err) {
  abort(err);
 }
}

function getBinaryPromise(binaryFile) {
 if (!wasmBinary && (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER)) {
  if (typeof fetch == "function" && !isFileURI(binaryFile)) {
   return fetch(binaryFile, {
    credentials: "same-origin"
   }).then(response => {
    if (!response["ok"]) {
     throw "failed to load wasm binary file at '" + binaryFile + "'";
    }
    return response["arrayBuffer"]();
   }).catch(() => getBinary(binaryFile));
  } else {
   if (readAsync) {
    return new Promise((resolve, reject) => {
     readAsync(binaryFile, response => resolve(new Uint8Array(response)), reject);
    });
   }
  }
 }
 return Promise.resolve().then(() => getBinary(binaryFile));
}

function instantiateArrayBuffer(binaryFile, imports, receiver) {
 return getBinaryPromise(binaryFile).then(binary => {
  return WebAssembly.instantiate(binary, imports);
 }).then(instance => {
  return instance;
 }).then(receiver, reason => {
  err("failed to asynchronously prepare wasm: " + reason);
  abort(reason);
 });
}

function instantiateAsync(binary, binaryFile, imports, callback) {
 if (!binary && typeof WebAssembly.instantiateStreaming == "function" && !isDataURI(binaryFile) && !isFileURI(binaryFile) && !ENVIRONMENT_IS_NODE && typeof fetch == "function") {
  return fetch(binaryFile, {
   credentials: "same-origin"
  }).then(response => {
   var result = WebAssembly.instantiateStreaming(response, imports);
   return result.then(callback, function(reason) {
    err("wasm streaming compile failed: " + reason);
    err("falling back to ArrayBuffer instantiation");
    return instantiateArrayBuffer(binaryFile, imports, callback);
   });
  });
 } else {
  return instantiateArrayBuffer(binaryFile, imports, callback);
 }
}

function createWasm() {
 var info = {
  "env": wasmImports,
  "wasi_snapshot_preview1": wasmImports
 };
 function receiveInstance(instance, module) {
  var exports = instance.exports;
  exports = instrumentWasmExportsForMemory64(exports);
  Module["asm"] = exports;
  wasmMemory = Module["asm"]["memory"];
  updateMemoryViews();
  wasmTable = Module["asm"]["__indirect_function_table"];
  addOnInit(Module["asm"]["__wasm_call_ctors"]);
  removeRunDependency("wasm-instantiate");
  return exports;
 }
 addRunDependency("wasm-instantiate");
 function receiveInstantiationResult(result) {
  receiveInstance(result["instance"]);
 }
 if (Module["instantiateWasm"]) {
  try {
   return Module["instantiateWasm"](info, receiveInstance);
  } catch (e) {
   err("Module.instantiateWasm callback failed with error: " + e);
   readyPromiseReject(e);
  }
 }
 instantiateAsync(wasmBinary, wasmBinaryFile, info, receiveInstantiationResult).catch(readyPromiseReject);
 return {};
}

function ExitStatus(status) {
 this.name = "ExitStatus";
 this.message = `Program terminated with exit(${status})`;
 this.status = status;
}

function callRuntimeCallbacks(callbacks) {
 while (callbacks.length > 0) {
  callbacks.shift()(Module);
 }
}

function getValue(ptr, type = "i8") {
 if (type.endsWith("*")) type = "*";
 switch (type) {
 case "i1":
  return HEAP8[ptr / 2 ** 0 >>> 0];

 case "i8":
  return HEAP8[ptr / 2 ** 0 >>> 0];

 case "i16":
  return HEAP16[ptr / 2 ** 1 >>> 0];

 case "i32":
  return HEAP32[ptr / 2 ** 2 >>> 0];

 case "i64":
  return HEAP64[ptr / 2 ** 3];

 case "float":
  return HEAPF32[ptr / 2 ** 2 >>> 0];

 case "double":
  return HEAPF64[ptr / 2 ** 3 >>> 0];

 case "*":
  return Number(HEAPU64[ptr / 2 ** 3]);

 default:
  abort(`invalid type for getValue: ${type}`);
 }
}

function setValue(ptr, value, type = "i8") {
 if (type.endsWith("*")) type = "*";
 switch (type) {
 case "i1":
  HEAP8[ptr / 2 ** 0 >>> 0] = value;
  break;

 case "i8":
  HEAP8[ptr / 2 ** 0 >>> 0] = value;
  break;

 case "i16":
  HEAP16[ptr / 2 ** 1 >>> 0] = value;
  break;

 case "i32":
  HEAP32[ptr / 2 ** 2 >>> 0] = value;
  break;

 case "i64":
  HEAP64[ptr / 2 ** 3] = BigInt(value);
  break;

 case "float":
  HEAPF32[ptr / 2 ** 2 >>> 0] = value;
  break;

 case "double":
  HEAPF64[ptr / 2 ** 3 >>> 0] = value;
  break;

 case "*":
  HEAPU64[ptr / 2 ** 3] = BigInt(value);
  break;

 default:
  abort(`invalid type for setValue: ${type}`);
 }
}

var exceptionCaught = [];

var uncaughtExceptionCount = 0;

function ___cxa_begin_catch(ptr) {
 var ret = (ptr => {
  var info = new ExceptionInfo(ptr);
  if (!info.get_caught()) {
   info.set_caught(true);
   uncaughtExceptionCount--;
  }
  info.set_rethrown(false);
  exceptionCaught.push(info);
  ___cxa_increment_exception_refcount(info.excPtr);
  return info.get_exception_ptr();
 })(Number(ptr));
 return BigInt(ret);
}

var exceptionLast = 0;

function ___cxa_end_catch() {
 _setThrew(0);
 var info = exceptionCaught.pop();
 ___cxa_decrement_exception_refcount(info.excPtr);
 exceptionLast = 0;
}

function ExceptionInfo(excPtr) {
 this.excPtr = excPtr;
 this.ptr = excPtr - 48;
 this.set_type = function(type) {
  HEAPU64[(this.ptr + 8) / 2 ** 3] = BigInt(type);
 };
 this.get_type = function() {
  return Number(HEAPU64[(this.ptr + 8) / 2 ** 3]);
 };
 this.set_destructor = function(destructor) {
  HEAPU64[(this.ptr + 16) / 2 ** 3] = BigInt(destructor);
 };
 this.get_destructor = function() {
  return Number(HEAPU64[(this.ptr + 16) / 2 ** 3]);
 };
 this.set_caught = function(caught) {
  caught = caught ? 1 : 0;
  HEAP8[(this.ptr + 24) / 2 ** 0 >>> 0] = caught;
 };
 this.get_caught = function() {
  return HEAP8[(this.ptr + 24) / 2 ** 0 >>> 0] != 0;
 };
 this.set_rethrown = function(rethrown) {
  rethrown = rethrown ? 1 : 0;
  HEAP8[(this.ptr + 25) / 2 ** 0 >>> 0] = rethrown;
 };
 this.get_rethrown = function() {
  return HEAP8[(this.ptr + 25) / 2 ** 0 >>> 0] != 0;
 };
 this.init = function(type, destructor) {
  this.set_adjusted_ptr(0);
  this.set_type(type);
  this.set_destructor(destructor);
 };
 this.set_adjusted_ptr = function(adjustedPtr) {
  HEAPU64[(this.ptr + 32) / 2 ** 3] = BigInt(adjustedPtr);
 };
 this.get_adjusted_ptr = function() {
  return Number(HEAPU64[(this.ptr + 32) / 2 ** 3]);
 };
 this.get_exception_ptr = function() {
  var isPointer = ___cxa_is_pointer_type(this.get_type());
  if (isPointer) {
   return Number(HEAPU64[this.excPtr / 2 ** 3]);
  }
  var adjusted = this.get_adjusted_ptr();
  if (adjusted !== 0) return adjusted;
  return this.excPtr;
 };
}

function ___resumeException(ptr) {
 ptr = Number(ptr);
 if (!exceptionLast) {
  exceptionLast = ptr;
 }
 throw exceptionLast;
}

function ___cxa_find_matching_catch() {
 var thrown = exceptionLast;
 if (!thrown) {
  setTempRet0(0);
  return BigInt(0);
 }
 var info = new ExceptionInfo(thrown);
 info.set_adjusted_ptr(thrown);
 var thrownType = info.get_type();
 if (!thrownType) {
  setTempRet0(0);
  return BigInt(thrown);
 }
 for (var i = 0; i < arguments.length; i++) {
  var caughtType = arguments[i];
  caughtType = Number(caughtType);
  if (caughtType === 0 || caughtType === thrownType) {
   break;
  }
  var adjusted_ptr_addr = info.ptr + 32;
  if (___cxa_can_catch(caughtType, thrownType, adjusted_ptr_addr)) {
   setTempRet0(caughtType);
   return BigInt(thrown);
  }
 }
 setTempRet0(thrownType);
 return BigInt(thrown);
}

var ___cxa_find_matching_catch_2 = ___cxa_find_matching_catch;

var ___cxa_find_matching_catch_3 = ___cxa_find_matching_catch;

var ___cxa_find_matching_catch_4 = ___cxa_find_matching_catch;

var ___cxa_find_matching_catch_5 = ___cxa_find_matching_catch;

function ___cxa_rethrow() {
 var info = exceptionCaught.pop();
 if (!info) {
  abort("no exception to throw");
 }
 var ptr = info.excPtr;
 if (!info.get_rethrown()) {
  exceptionCaught.push(info);
  info.set_rethrown(true);
  info.set_caught(false);
  uncaughtExceptionCount++;
 }
 exceptionLast = ptr;
 throw exceptionLast;
}

function ___cxa_throw(ptr, type, destructor) {
 ptr = Number(ptr);
 type = Number(type);
 destructor = Number(destructor);
 var info = new ExceptionInfo(ptr);
 info.init(type, destructor);
 exceptionLast = ptr;
 uncaughtExceptionCount++;
 throw exceptionLast;
}

function ___cxa_uncaught_exceptions() {
 return uncaughtExceptionCount;
}

var dlopenMissingError = "To use dlopen, you need enable dynamic linking, see https://emscripten.org/docs/compiling/Dynamic-Linking.html";

function ___dlsym(handle, symbol, ra) {
 var ret = ((handle, symbol, ra) => {
  abort(dlopenMissingError);
 })(Number(handle), Number(symbol), Number(ra));
 return BigInt(ret);
}

function setErrNo(value) {
 HEAP32[___errno_location() / 2 ** 2 >>> 0] = value;
 return value;
}

var UTF8Decoder = typeof TextDecoder != "undefined" ? new TextDecoder("utf8") : undefined;

function fixPointer (idx) {
 if (idx < 0) {
  return idx >>> 0
 }

 return idx;
}

function UTF8ArrayToString(heapOrArray, idx, maxBytesToRead) {
 idx = fixPointer(idx);
 var endIdx = idx + maxBytesToRead;
 var endPtr = idx;
 while (heapOrArray[endPtr] && !(endPtr >= endIdx)) ++endPtr;
 if (endPtr - idx > 16 && heapOrArray.buffer && UTF8Decoder) {
  return UTF8Decoder.decode(heapOrArray.subarray(idx, endPtr));
 }
 var str = "";
 while (idx < endPtr) {
  var u0 = heapOrArray[idx++];
  if (!(u0 & 128)) {
   str += String.fromCharCode(u0);
   continue;
  }
  var u1 = heapOrArray[idx++] & 63;
  if ((u0 & 224) == 192) {
   str += String.fromCharCode((u0 & 31) << 6 | u1);
   continue;
  }
  var u2 = heapOrArray[idx++] & 63;
  if ((u0 & 240) == 224) {
   u0 = (u0 & 15) << 12 | u1 << 6 | u2;
  } else {
   u0 = (u0 & 7) << 18 | u1 << 12 | u2 << 6 | heapOrArray[idx++] & 63;
  }
  if (u0 < 65536) {
   str += String.fromCharCode(u0);
  } else {
   var ch = u0 - 65536;
   str += String.fromCharCode(55296 | ch >> 10, 56320 | ch & 1023);
  }
 }
 return str;
}

function UTF8ToString(ptr, maxBytesToRead) {
 if (typeof ptr !== "bigint") {
  ptr = fixPointer(ptr);
 } else {
  ptr = Number(ptr);
 }
 return ptr ? UTF8ArrayToString(HEAPU8, ptr, maxBytesToRead) : "";
}

var SYSCALLS = {
 varargs: undefined,
 get: function() {
  SYSCALLS.varargs += 4;
  var ret = HEAP32[(SYSCALLS.varargs - 4) / 2 ** 2 >>> 0];
  return ret;
 },
 getStr: function(ptr) {
  var ret = UTF8ToString(ptr);
  return ret;
 }
};

function ___syscall_fcntl64(fd, cmd, varargs) {
 varargs = Number(varargs);
 SYSCALLS.varargs = varargs;
 return 0;
}

function ___syscall_fstat64(fd, buf) {
 buf = Number(buf);
}

function lengthBytesUTF8(str) {
 var len = 0;
 for (var i = 0; i < str.length; ++i) {
  var c = str.charCodeAt(i);
  if (c <= 127) {
   len++;
  } else if (c <= 2047) {
   len += 2;
  } else if (c >= 55296 && c <= 57343) {
   len += 4;
   ++i;
  } else {
   len += 3;
  }
 }
 return len;
}

function stringToUTF8Array(str, heap, outIdx, maxBytesToWrite) {
 outIdx = fixPointer(outIdx);
 if (!(maxBytesToWrite > 0)) return 0;
 var startIdx = outIdx;
 var endIdx = outIdx + maxBytesToWrite - 1;
 for (var i = 0; i < str.length; ++i) {
  var u = str.charCodeAt(i);
  if (u >= 55296 && u <= 57343) {
   var u1 = str.charCodeAt(++i);
   u = 65536 + ((u & 1023) << 10) | u1 & 1023;
  }
  if (u <= 127) {
   if (outIdx >= endIdx) break;
   heap[outIdx++] = u;
  } else if (u <= 2047) {
   if (outIdx + 1 >= endIdx) break;
   heap[outIdx++] = 192 | u >> 6;
   heap[outIdx++] = 128 | u & 63;
  } else if (u <= 65535) {
   if (outIdx + 2 >= endIdx) break;
   heap[outIdx++] = 224 | u >> 12;
   heap[outIdx++] = 128 | u >> 6 & 63;
   heap[outIdx++] = 128 | u & 63;
  } else {
   if (outIdx + 3 >= endIdx) break;
   heap[outIdx++] = 240 | u >> 18;
   heap[outIdx++] = 128 | u >> 12 & 63;
   heap[outIdx++] = 128 | u >> 6 & 63;
   heap[outIdx++] = 128 | u & 63;
  }
 }
 heap[outIdx] = 0;
 return outIdx - startIdx;
}

function stringToUTF8(str, outPtr, maxBytesToWrite) {
 return stringToUTF8Array(str, HEAPU8, outPtr, maxBytesToWrite);
}

function ___syscall_getcwd(buf, size) {
 buf = Number(buf);
 size = Number(size);
}

function ___syscall_getdents64(fd, dirp, count) {
 dirp = Number(dirp);
 count = Number(count);
}

function ___syscall_ioctl(fd, op, varargs) {
 varargs = Number(varargs);
 SYSCALLS.varargs = varargs;
 return 0;
}

function ___syscall_lstat64(path, buf) {
 path = Number(path);
 buf = Number(buf);
}

function ___syscall_mkdirat(dirfd, path, mode) {
 path = Number(path);
}

function ___syscall_newfstatat(dirfd, path, buf, flags) {
 path = Number(path);
 buf = Number(buf);
}

function ___syscall_openat(dirfd, path, flags, varargs) {
 path = Number(path);
 varargs = Number(varargs);
 SYSCALLS.varargs = varargs;
}

function ___syscall_readlinkat(dirfd, path, buf, bufsize) {
 path = Number(path);
 buf = Number(buf);
 bufsize = Number(bufsize);
}

function ___syscall_rmdir(path) {
 path = Number(path);
}

function ___syscall_stat64(path, buf) {
 path = Number(path);
 buf = Number(buf);
}

function ___syscall_unlinkat(dirfd, path, flags) {
 path = Number(path);
}

function embindRepr(v) {
 if (v === null) {
  return "null";
 }
 var t = typeof v;
 if (t === "object" || t === "array" || t === "function") {
  return v.toString();
 } else {
  return "" + v;
 }
}

function embind_init_charCodes() {
 var codes = new Array(256);
 for (var i = 0; i < 256; ++i) {
  codes[i] = String.fromCharCode(i);
 }
 embind_charCodes = codes;
}

var embind_charCodes = undefined;

function readLatin1String(ptr) {
 var ret = "";
 var c = ptr;
 while (HEAPU8[c >>> 0]) {
  ret += embind_charCodes[HEAPU8[c++ >>> 0]];
 }
 return ret;
}

var awaitingDependencies = {};

var registeredTypes = {};

var typeDependencies = {};

var char_0 = 48;

var char_9 = 57;

function makeLegalFunctionName(name) {
 if (undefined === name) {
  return "_unknown";
 }
 name = name.replace(/[^a-zA-Z0-9_]/g, "$");
 var f = name.charCodeAt(0);
 if (f >= char_0 && f <= char_9) {
  return `_${name}`;
 }
 return name;
}

function createNamedFunction(name, body) {
 name = makeLegalFunctionName(name);
 return {
  [name]: function() {
   return body.apply(this, arguments);
  }
 }[name];
}

function extendError(baseErrorType, errorName) {
 var errorClass = createNamedFunction(errorName, function(message) {
  this.name = errorName;
  this.message = message;
  var stack = new Error(message).stack;
  if (stack !== undefined) {
   this.stack = this.toString() + "\n" + stack.replace(/^Error(:[^\n]*)?\n/, "");
  }
 });
 errorClass.prototype = Object.create(baseErrorType.prototype);
 errorClass.prototype.constructor = errorClass;
 errorClass.prototype.toString = function() {
  if (this.message === undefined) {
   return this.name;
  } else {
   return `${this.name}: ${this.message}`;
  }
 };
 return errorClass;
}

var BindingError = undefined;

function throwBindingError(message) {
 throw new BindingError(message);
}

var InternalError = undefined;

function throwInternalError(message) {
 throw new InternalError(message);
}

function whenDependentTypesAreResolved(myTypes, dependentTypes, getTypeConverters) {
 myTypes.forEach(function(type) {
  typeDependencies[type] = dependentTypes;
 });
 function onComplete(typeConverters) {
  var myTypeConverters = getTypeConverters(typeConverters);
  if (myTypeConverters.length !== myTypes.length) {
   throwInternalError("Mismatched type converter count");
  }
  for (var i = 0; i < myTypes.length; ++i) {
   registerType(myTypes[i], myTypeConverters[i]);
  }
 }
 var typeConverters = new Array(dependentTypes.length);
 var unregisteredTypes = [];
 var registered = 0;
 dependentTypes.forEach((dt, i) => {
  if (registeredTypes.hasOwnProperty(dt)) {
   typeConverters[i] = registeredTypes[dt];
  } else {
   unregisteredTypes.push(dt);
   if (!awaitingDependencies.hasOwnProperty(dt)) {
    awaitingDependencies[dt] = [];
   }
   awaitingDependencies[dt].push(() => {
    typeConverters[i] = registeredTypes[dt];
    ++registered;
    if (registered === unregisteredTypes.length) {
     onComplete(typeConverters);
    }
   });
  }
 });
 if (0 === unregisteredTypes.length) {
  onComplete(typeConverters);
 }
}

function registerType(rawType, registeredInstance, options = {}) {
 if (!("argPackAdvance" in registeredInstance)) {
  throw new TypeError("registerType registeredInstance requires argPackAdvance");
 }
 var name = registeredInstance.name;
 if (!rawType) {
  throwBindingError(`type "${name}" must have a positive integer typeid pointer`);
 }
 if (registeredTypes.hasOwnProperty(rawType)) {
  if (options.ignoreDuplicateRegistrations) {
   return;
  } else {
   throwBindingError(`Cannot register type '${name}' twice`);
  }
 }
 registeredTypes[rawType] = registeredInstance;
 delete typeDependencies[rawType];
 if (awaitingDependencies.hasOwnProperty(rawType)) {
  var callbacks = awaitingDependencies[rawType];
  delete awaitingDependencies[rawType];
  callbacks.forEach(cb => cb());
 }
}

function integerReadValueFromPointer(name, shift, signed) {
 switch (shift) {
 case 0:
  return signed ? function readS8FromPointer(pointer) {
   return HEAP8[pointer >>> 0];
  } : function readU8FromPointer(pointer) {
   return HEAPU8[pointer >>> 0];
  };

 case 1:
  return signed ? function readS16FromPointer(pointer) {
   return HEAP16[pointer >>> 1];
  } : function readU16FromPointer(pointer) {
   return HEAPU16[pointer >>> 1];
  };

 case 2:
  return signed ? function readS32FromPointer(pointer) {
   return HEAP32[pointer >>> 2];
  } : function readU32FromPointer(pointer) {
   return HEAPU32[pointer >>> 2];
  };

 case 3:
  return signed ? function readS64FromPointer(pointer) {
   return HEAP64[pointer >> 3];
  } : function readU64FromPointer(pointer) {
   return HEAPU64[pointer >> 3];
  };

 default:
  throw new TypeError("Unknown integer type: " + name);
 }
}

function __embind_register_bigint(primitiveType, name, size, minRange, maxRange) {
 primitiveType = Number(primitiveType);
 name = Number(name);
 size = Number(size);
 name = readLatin1String(name);
 var shift = getShiftFromSize(size);
 var isUnsignedType = name.indexOf("u") != -1;
 if (isUnsignedType) {
  maxRange = (1n << 64n) - 1n;
 }
 registerType(primitiveType, {
  name: name,
  "fromWireType": function(value) {
   return value;
  },
  "toWireType": function(destructors, value) {
   if (typeof value != "bigint" && typeof value != "number") {
    throw new TypeError(`Cannot convert "${embindRepr(value)}" to ${this.name}`);
   }
   if (value < minRange || value > maxRange) {
    throw new TypeError(`Passing a number "${embindRepr(value)}" from JS side to C/C++ side to an argument of type "${name}", which is outside the valid range [${minRange}, ${maxRange}]!`);
   }
   return value;
  },
  "argPackAdvance": 8,
  "readValueFromPointer": integerReadValueFromPointer(name, shift, !isUnsignedType),
  destructorFunction: null
 });
}

function getShiftFromSize(size) {
 switch (size) {
 case 1:
  return 0;

 case 2:
  return 1;

 case 4:
  return 2;

 case 8:
  return 3;

 default:
  throw new TypeError(`Unknown type size: ${size}`);
 }
}

function __embind_register_bool(rawType, name, size, trueValue, falseValue) {
 rawType = Number(rawType);
 name = Number(name);
 size = Number(size);
 var shift = getShiftFromSize(size);
 name = readLatin1String(name);
 registerType(rawType, {
  name: name,
  "fromWireType": function(wt) {
   return !!wt;
  },
  "toWireType": function(destructors, o) {
   return o ? trueValue : falseValue;
  },
  "argPackAdvance": 8,
  "readValueFromPointer": function(pointer) {
   var heap;
   if (size === 1) {
    heap = HEAP8;
   } else if (size === 2) {
    heap = HEAP16;
   } else if (size === 4) {
    heap = HEAP32;
   } else {
    throw new TypeError("Unknown boolean type size: " + name);
   }
   return this["fromWireType"](heap[pointer >>> shift]);
  },
  destructorFunction: null
 });
}

function HandleAllocator() {
 this.allocated = [ undefined ];
 this.freelist = [];
 this.get = function(id) {
  return this.allocated[id];
 };
 this.has = function(id) {
  return this.allocated[id] !== undefined;
 };
 this.allocate = function(handle) {
  var id = this.freelist.pop() || this.allocated.length;
  this.allocated[id] = handle;
  return id;
 };
 this.free = function(id) {
  this.allocated[id] = undefined;
  this.freelist.push(id);
 };
}

var emval_handles = new HandleAllocator();

function __emval_decref(handle) {
 handle = Number(handle);
 if (handle >= emval_handles.reserved && 0 === --emval_handles.get(handle).refcount) {
  emval_handles.free(handle);
 }
}

function count_emval_handles() {
 var count = 0;
 for (var i = emval_handles.reserved; i < emval_handles.allocated.length; ++i) {
  if (emval_handles.allocated[i] !== undefined) {
   ++count;
  }
 }
 return count;
}

function init_emval() {
 emval_handles.allocated.push({
  value: undefined
 }, {
  value: null
 }, {
  value: true
 }, {
  value: false
 });
 emval_handles.reserved = emval_handles.allocated.length;
 Module["count_emval_handles"] = count_emval_handles;
}

var Emval = {
 toValue: handle => {
  if (!handle) {
   throwBindingError("Cannot use deleted val. handle = " + handle);
  }
  return emval_handles.get(handle).value;
 },
 toHandle: value => {
  switch (value) {
  case undefined:
   return 1;

  case null:
   return 2;

  case true:
   return 3;

  case false:
   return 4;

  default:
   {
    return emval_handles.allocate({
     refcount: 1,
     value: value
    });
   }
  }
 }
};

function simpleReadValueFromPointer(pointer) {
 return this["fromWireType"](HEAP32[pointer / 2 ** 2 >>> 0]);
}

function __embind_register_emval(rawType, name) {
 rawType = Number(rawType);
 name = Number(name);
 name = readLatin1String(name);
 registerType(rawType, {
  name: name,
  "fromWireType": function(handle) {
   var rv = Emval.toValue(handle);
   __emval_decref(handle);
   return rv;
  },
  "toWireType": function(destructors, value) {
   return Emval.toHandle(value);
  },
  "argPackAdvance": 8,
  "readValueFromPointer": simpleReadValueFromPointer,
  destructorFunction: null
 });
}

function floatReadValueFromPointer(name, shift) {
 switch (shift) {
 case 2:
  return function(pointer) {
   return this["fromWireType"](HEAPF32[pointer >>> 2]);
  };

 case 3:
  return function(pointer) {
   return this["fromWireType"](HEAPF64[pointer >>> 3]);
  };

 default:
  throw new TypeError("Unknown float type: " + name);
 }
}

function __embind_register_float(rawType, name, size) {
 rawType = Number(rawType);
 name = Number(name);
 size = Number(size);
 var shift = getShiftFromSize(size);
 name = readLatin1String(name);
 registerType(rawType, {
  name: name,
  "fromWireType": function(value) {
   return value;
  },
  "toWireType": function(destructors, value) {
   return value;
  },
  "argPackAdvance": 8,
  "readValueFromPointer": floatReadValueFromPointer(name, shift),
  destructorFunction: null
 });
}

function __embind_register_integer(primitiveType, name, size, minRange, maxRange) {
 primitiveType = Number(primitiveType);
 name = Number(name);
 size = Number(size);
 name = readLatin1String(name);
 if (maxRange === -1) {
  maxRange = 4294967295;
 }
 var shift = getShiftFromSize(size);
 var fromWireType = value => value;
 if (minRange === 0) {
  var bitshift = 32 - 8 * size;
  fromWireType = value => value << bitshift >>> bitshift;
 }
 var isUnsignedType = name.includes("unsigned");
 var checkAssertions = (value, toTypeName) => {};
 var toWireType;
 if (isUnsignedType) {
  toWireType = function(destructors, value) {
   checkAssertions(value, this.name);
   return value >>> 0;
  };
 } else {
  toWireType = function(destructors, value) {
   checkAssertions(value, this.name);
   return value;
  };
 }
 registerType(primitiveType, {
  name: name,
  "fromWireType": fromWireType,
  "toWireType": toWireType,
  "argPackAdvance": 8,
  "readValueFromPointer": integerReadValueFromPointer(name, shift, minRange !== 0),
  destructorFunction: null
 });
}

function __embind_register_memory_view(rawType, dataTypeIndex, name) {
 rawType = Number(rawType);
 name = Number(name);
 var typeMapping = [ Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array, BigInt64Array, BigUint64Array ];
 var TA = typeMapping[dataTypeIndex];
 function decodeMemoryView(handle) {
  handle = handle >> 2;
  var heap = HEAPU32;
  var size = heap[handle >>> 0];
  var data = heap[handle + 1 >>> 0];
  return new TA(heap.buffer, data, size);
 }
 name = readLatin1String(name);
 registerType(rawType, {
  name: name,
  "fromWireType": decodeMemoryView,
  "argPackAdvance": 8,
  "readValueFromPointer": decodeMemoryView
 }, {
  ignoreDuplicateRegistrations: true
 });
}

function __embind_register_std_string(rawType, name) {
 rawType = Number(rawType);
 name = Number(name);
 name = readLatin1String(name);
 var stdStringIsUTF8 = name === "std::string";
 registerType(rawType, {
  name: name,
  "fromWireType": function(value) {
   var length = HEAPU64[value / 2 ** 3];
   var payload = value + 8;
   var str;
   if (stdStringIsUTF8) {
    var decodeStartPtr = payload;
    for (var i = 0; i <= length; ++i) {
     var currentBytePtr = payload + i;
     if (i == length || HEAPU8[currentBytePtr >>> 0] == 0) {
      var maxRead = currentBytePtr - decodeStartPtr;
      var stringSegment = UTF8ToString(decodeStartPtr, maxRead);
      if (str === undefined) {
       str = stringSegment;
      } else {
       str += String.fromCharCode(0);
       str += stringSegment;
      }
      decodeStartPtr = currentBytePtr + 1;
     }
    }
   } else {
    var a = new Array(length);
    for (var i = 0; i < length; ++i) {
     a[i] = String.fromCharCode(HEAPU8[payload + i >>> 0]);
    }
    str = a.join("");
   }
   _free(value);
   return str;
  },
  "toWireType": function(destructors, value) {
   if (value instanceof ArrayBuffer) {
    value = new Uint8Array(value);
   }
   var length;
   var valueIsOfTypeString = typeof value == "string";
   if (!(valueIsOfTypeString || value instanceof Uint8Array || value instanceof Uint8ClampedArray || value instanceof Int8Array)) {
    throwBindingError("Cannot pass non-string to std::string");
   }
   if (stdStringIsUTF8 && valueIsOfTypeString) {
    length = lengthBytesUTF8(value);
   } else {
    length = value.length;
   }
   var base = _malloc(8 + length + 1);
   var ptr = base + 8;
   ptr >>>= 0;
   HEAPU64[base / 2 ** 3] = BigInt(length);
   if (stdStringIsUTF8 && valueIsOfTypeString) {
    stringToUTF8(value, ptr, length + 1);
   } else {
    if (valueIsOfTypeString) {
     for (var i = 0; i < length; ++i) {
      var charCode = value.charCodeAt(i);
      if (charCode > 255) {
       _free(ptr);
       throwBindingError("String has UTF-16 code units that do not fit in 8 bits");
      }
      HEAPU8[ptr + i >>> 0] = charCode;
     }
    } else {
     for (var i = 0; i < length; ++i) {
      HEAPU8[ptr + i >>> 0] = value[i];
     }
    }
   }
   if (destructors !== null) {
    destructors.push(_free, base);
   }
   return base;
  },
  "argPackAdvance": 8,
  "readValueFromPointer": simpleReadValueFromPointer,
  destructorFunction: function(ptr) {
   _free(ptr);
  }
 });
}

var UTF16Decoder = typeof TextDecoder != "undefined" ? new TextDecoder("utf-16le") : undefined;

function UTF16ToString(ptr, maxBytesToRead) {
 var endPtr = ptr;
 var idx = endPtr >> 1;
 var maxIdx = idx + maxBytesToRead / 2;
 while (!(idx >= maxIdx) && HEAPU16[idx >>> 0]) ++idx;
 endPtr = idx << 1;
 if (endPtr - ptr > 32 && UTF16Decoder) return UTF16Decoder.decode(HEAPU8.subarray(ptr >>> 0, endPtr >>> 0));
 var str = "";
 for (var i = 0; !(i >= maxBytesToRead / 2); ++i) {
  var codeUnit = HEAP16[(ptr + i * 2) / 2 ** 1 >>> 0];
  if (codeUnit == 0) break;
  str += String.fromCharCode(codeUnit);
 }
 return str;
}

function stringToUTF16(str, outPtr, maxBytesToWrite) {
 if (maxBytesToWrite === undefined) {
  maxBytesToWrite = 2147483647;
 }
 if (maxBytesToWrite < 2) return 0;
 maxBytesToWrite -= 2;
 var startPtr = outPtr;
 var numCharsToWrite = maxBytesToWrite < str.length * 2 ? maxBytesToWrite / 2 : str.length;
 for (var i = 0; i < numCharsToWrite; ++i) {
  var codeUnit = str.charCodeAt(i);
  HEAP16[outPtr / 2 ** 1 >>> 0] = codeUnit;
  outPtr += 2;
 }
 HEAP16[outPtr / 2 ** 1 >>> 0] = 0;
 return outPtr - startPtr;
}

function lengthBytesUTF16(str) {
 return str.length * 2;
}

function UTF32ToString(ptr, maxBytesToRead) {
 var i = 0;
 var str = "";
 while (!(i >= maxBytesToRead / 4)) {
  var utf32 = HEAP32[(ptr + i * 4) / 2 ** 2 >>> 0];
  if (utf32 == 0) break;
  ++i;
  if (utf32 >= 65536) {
   var ch = utf32 - 65536;
   str += String.fromCharCode(55296 | ch >> 10, 56320 | ch & 1023);
  } else {
   str += String.fromCharCode(utf32);
  }
 }
 return str;
}

function stringToUTF32(str, outPtr, maxBytesToWrite) {
 outPtr >>>= 0;
 if (maxBytesToWrite === undefined) {
  maxBytesToWrite = 2147483647;
 }
 if (maxBytesToWrite < 4) return 0;
 var startPtr = outPtr;
 var endPtr = startPtr + maxBytesToWrite - 4;
 for (var i = 0; i < str.length; ++i) {
  var codeUnit = str.charCodeAt(i);
  if (codeUnit >= 55296 && codeUnit <= 57343) {
   var trailSurrogate = str.charCodeAt(++i);
   codeUnit = 65536 + ((codeUnit & 1023) << 10) | trailSurrogate & 1023;
  }
  HEAP32[outPtr / 2 ** 2 >>> 0] = codeUnit;
  outPtr += 4;
  if (outPtr + 4 > endPtr) break;
 }
 HEAP32[outPtr / 2 ** 2 >>> 0] = 0;
 return outPtr - startPtr;
}

function lengthBytesUTF32(str) {
 var len = 0;
 for (var i = 0; i < str.length; ++i) {
  var codeUnit = str.charCodeAt(i);
  if (codeUnit >= 55296 && codeUnit <= 57343) ++i;
  len += 4;
 }
 return len;
}

function __embind_register_std_wstring(rawType, charSize, name) {
 rawType = Number(rawType);
 charSize = Number(charSize);
 name = Number(name);
 name = readLatin1String(name);
 var decodeString, encodeString, getHeap, lengthBytesUTF, shift;
 if (charSize === 2) {
  decodeString = UTF16ToString;
  encodeString = stringToUTF16;
  lengthBytesUTF = lengthBytesUTF16;
  getHeap = () => HEAPU16;
  shift = 1;
 } else if (charSize === 4) {
  decodeString = UTF32ToString;
  encodeString = stringToUTF32;
  lengthBytesUTF = lengthBytesUTF32;
  getHeap = () => HEAPU32;
  shift = 2;
 }
 registerType(rawType, {
  name: name,
  "fromWireType": function(value) {
   var length = HEAPU32[value >>> 2];
   var HEAP = getHeap();
   var str;
   var decodeStartPtr = value + 4;
   for (var i = 0; i <= length; ++i) {
    var currentBytePtr = value + 4 + i * charSize;
    if (i == length || HEAP[currentBytePtr >>> shift] == 0) {
     var maxReadBytes = currentBytePtr - decodeStartPtr;
     var stringSegment = decodeString(decodeStartPtr, maxReadBytes);
     if (str === undefined) {
      str = stringSegment;
     } else {
      str += String.fromCharCode(0);
      str += stringSegment;
     }
     decodeStartPtr = currentBytePtr + charSize;
    }
   }
   _free(value);
   return str;
  },
  "toWireType": function(destructors, value) {
   if (!(typeof value == "string")) {
    throwBindingError(`Cannot pass non-string to C++ string type ${name}`);
   }
   var length = lengthBytesUTF(value);
   var ptr = _malloc(4 + length + charSize);
   ptr >>>= 0;
   HEAPU32[ptr >>> 2] = length >> shift;
   encodeString(value, ptr + 4, length + charSize);
   if (destructors !== null) {
    destructors.push(_free, ptr);
   }
   return ptr;
  },
  "argPackAdvance": 8,
  "readValueFromPointer": simpleReadValueFromPointer,
  destructorFunction: function(ptr) {
   _free(ptr);
  }
 });
}

function __embind_register_void(rawType, name) {
 rawType = Number(rawType);
 name = Number(name);
 name = readLatin1String(name);
 registerType(rawType, {
  isVoid: true,
  name: name,
  "argPackAdvance": 0,
  "fromWireType": function() {
   return undefined;
  },
  "toWireType": function(destructors, o) {
   return undefined;
  }
 });
}

var nowIsMonotonic = true;

function __emscripten_get_now_is_monotonic() {
 return nowIsMonotonic;
}

function getTypeName(type) {
 var ptr = ___getTypeName(type);
 var rv = readLatin1String(ptr);
 _free(ptr);
 return rv;
}

function requireRegisteredType(rawType, humanName) {
 var impl = registeredTypes[rawType];
 if (undefined === impl) {
  throwBindingError(humanName + " has unknown type " + getTypeName(rawType));
 }
 return impl;
}

function __emval_as(handle, returnType, destructorsRef) {
 handle = Number(handle);
 returnType = Number(returnType);
 destructorsRef = Number(destructorsRef);
 handle = Emval.toValue(handle);
 returnType = requireRegisteredType(returnType, "emval::as");
 var destructors = [];
 var rd = Emval.toHandle(destructors);
 HEAPU64[destructorsRef / 2 ** 3] = BigInt(rd);
 return returnType["toWireType"](destructors, handle);
}

function emval_allocateDestructors(destructorsRef) {
 var destructors = [];
 HEAPU64[destructorsRef / 2 ** 3] = BigInt(Emval.toHandle(destructors));
 return destructors;
}

var emval_symbols = {};

function getStringOrSymbol(address) {
 var symbol = emval_symbols[address];
 if (symbol === undefined) {
  return readLatin1String(address);
 }
 return symbol;
}

var emval_methodCallers = [];

function __emval_call_method(caller, handle, methodName, destructorsRef, args) {
 caller = Number(caller);
 handle = Number(handle);
 methodName = Number(methodName);
 destructorsRef = Number(destructorsRef);
 args = Number(args);
 caller = emval_methodCallers[caller];
 handle = Emval.toValue(handle);
 methodName = getStringOrSymbol(methodName);
 return caller(handle, methodName, emval_allocateDestructors(destructorsRef), args);
}

function __emval_call_void_method(caller, handle, methodName, args) {
 caller = Number(caller);
 handle = Number(handle);
 methodName = Number(methodName);
 args = Number(args);
 caller = emval_methodCallers[caller];
 handle = Emval.toValue(handle);
 methodName = getStringOrSymbol(methodName);
 caller(handle, methodName, null, args);
}

function __emval_equals(first, second) {
 first = Number(first);
 second = Number(second);
 first = Emval.toValue(first);
 second = Emval.toValue(second);
 return first == second;
}

function emval_get_global() {
 if (typeof globalThis == "object") {
  return globalThis;
 }
 return function() {
  return Function;
 }()("return this")();
}

function __emval_get_global(name) {
 var ret = (name => {
  if (name === 0) {
   return Emval.toHandle(emval_get_global());
  } else {
   name = getStringOrSymbol(name);
   return Emval.toHandle(emval_get_global()[name]);
  }
 })(Number(name));
 return BigInt(ret);
}

function emval_addMethodCaller(caller) {
 var id = emval_methodCallers.length;
 emval_methodCallers.push(caller);
 return id;
}

function emval_lookupTypes(argCount, argTypes) {
 var a = new Array(argCount);
 for (var i = 0; i < argCount; ++i) {
  a[i] = requireRegisteredType(Number(HEAPU64[(argTypes + i * 8) / 2 ** 3]), "parameter " + i);
 }
 return a;
}

var emval_registeredMethods = [];

function newFunc(constructor, argumentList) {
 if (!(constructor instanceof Function)) {
  throw new TypeError(`new_ called with constructor type ${typeof constructor} which is not a function`);
 }
 var dummy = createNamedFunction(constructor.name || "unknownFunctionName", function() {});
 dummy.prototype = constructor.prototype;
 var obj = new dummy();
 var r = constructor.apply(obj, argumentList);
 return r instanceof Object ? r : obj;
}

function __emval_get_method_caller(argCount, argTypes) {
 var ret = ((argCount, argTypes) => {
  var types = emval_lookupTypes(argCount, argTypes);
  var retType = types[0];
  var signatureName = retType.name + "_$" + types.slice(1).map(function(t) {
   return t.name;
  }).join("_") + "$";
  var returnId = emval_registeredMethods[signatureName];
  if (returnId !== undefined) {
   return returnId;
  }
  var params = [ "retType" ];
  var args = [ retType ];
  var argsList = "";
  for (var i = 0; i < argCount - 1; ++i) {
   argsList += (i !== 0 ? ", " : "") + "arg" + i;
   params.push("argType" + i);
   args.push(types[1 + i]);
  }
  var functionName = makeLegalFunctionName("methodCaller_" + signatureName);
  var functionBody = "return function " + functionName + "(handle, name, destructors, args) {\n";
  var offset = 0;
  for (var i = 0; i < argCount - 1; ++i) {
   functionBody += "    var arg" + i + " = argType" + i + ".readValueFromPointer(args" + (offset ? "+" + offset : "") + ");\n";
   offset += types[i + 1]["argPackAdvance"];
  }
  functionBody += "    var rv = handle[name](" + argsList + ");\n";
  for (var i = 0; i < argCount - 1; ++i) {
   if (types[i + 1]["deleteObject"]) {
    functionBody += "    argType" + i + ".deleteObject(arg" + i + ");\n";
   }
  }
  if (!retType.isVoid) {
   functionBody += "    return retType.toWireType(destructors, rv);\n";
  }
  functionBody += "};\n";
  params.push(functionBody);
  var invokerFunction = newFunc(Function, params).apply(null, args);
  returnId = emval_addMethodCaller(invokerFunction);
  emval_registeredMethods[signatureName] = returnId;
  return returnId;
 })(argCount, Number(argTypes));
 return BigInt(ret);
}

function __emval_get_property(handle, key) {
 var ret = ((handle, key) => {
  handle = Emval.toValue(handle);
  key = Emval.toValue(key);
  return Emval.toHandle(handle[key]);
 })(Number(handle), Number(key));
 return BigInt(ret);
}

function __emval_incref(handle) {
 handle = Number(handle);
 if (handle > 4) {
  emval_handles.get(handle).refcount += 1;
 }
}

function craftEmvalAllocator(argCount) {
 var argsList = "";
 for (var i = 0; i < argCount; ++i) {
  argsList += (i !== 0 ? ", " : "") + "arg" + i;
 }
 var getMemory = () => HEAPU64;
 var functionBody = "return function emval_allocator_" + argCount + "(constructor, argTypes, args) {\n" + "  var HEAPU64 = getMemory();\n";
 for (var i = 0; i < argCount; ++i) {
  functionBody += "var argType" + i + " = requireRegisteredType(Number(HEAPU64[((argTypes)/2**3)]), 'parameter " + i + "');\n" + "var arg" + i + " = argType" + i + ".readValueFromPointer(args);\n" + "args += argType" + i + "['argPackAdvance'];\n" + "argTypes += 8;\n";
 }
 functionBody += "var obj = new constructor(" + argsList + ");\n" + "return valueToHandle(obj);\n" + "}\n";
 return new Function("requireRegisteredType", "Module", "valueToHandle", "getMemory", functionBody)(requireRegisteredType, Module, Emval.toHandle, getMemory);
}

var emval_newers = {};

function __emval_new(handle, argCount, argTypes, args) {
 var ret = ((handle, argCount, argTypes, args) => {
  handle = Emval.toValue(handle);
  var newer = emval_newers[argCount];
  if (!newer) {
   newer = craftEmvalAllocator(argCount);
   emval_newers[argCount] = newer;
  }
  return newer(handle, argTypes, args);
 })(Number(handle), argCount, Number(argTypes), Number(args));
 return BigInt(ret);
}

function __emval_new_array() {
 var ret = (() => {
  return Emval.toHandle([]);
 })();
 return BigInt(ret);
}

function __emval_new_array_from_memory_view(view) {
 var ret = (view => {
  view = Emval.toValue(view);
  var a = new Array(view.length);
  for (var i = 0; i < view.length; i++) a[i] = view[i];
  return Emval.toHandle(a);
 })(Number(view));
 return BigInt(ret);
}

function __emval_new_cstring(v) {
 var ret = (v => {
  return Emval.toHandle(getStringOrSymbol(v));
 })(Number(v));
 return BigInt(ret);
}

function __emval_new_object() {
 var ret = (() => {
  return Emval.toHandle({});
 })();
 return BigInt(ret);
}

function runDestructors(destructors) {
 while (destructors.length) {
  var ptr = destructors.pop();
  var del = destructors.pop();
  del(ptr);
 }
}

function __emval_run_destructors(handle) {
 handle = Number(handle);
 var destructors = Emval.toValue(handle);
 runDestructors(destructors);
 __emval_decref(handle);
}

function __emval_set_property(handle, key, value) {
 handle = Number(handle);
 key = Number(key);
 value = Number(value);
 handle = Emval.toValue(handle);
 key = Emval.toValue(key);
 value = Emval.toValue(value);
 handle[key] = value;
}

function __emval_take_value(type, arg) {
 var ret = ((type, arg) => {
  type = requireRegisteredType(type, "_emval_take_value");
  var v = type["readValueFromPointer"](arg);
  return Emval.toHandle(v);
 })(Number(type), Number(arg));
 return BigInt(ret);
}

function readI53FromI64(ptr) {
 return HEAPU32[ptr >>> 2] + HEAP32[ptr + 4 >>> 2] * 4294967296;
}

function __gmtime_js(time, tmPtr) {
 time = Number(time);
 tmPtr = Number(tmPtr);
 var date = new Date(readI53FromI64(time) * 1e3);
 HEAP32[tmPtr / 2 ** 2 >>> 0] = date.getUTCSeconds();
 HEAP32[(tmPtr + 4) / 2 ** 2 >>> 0] = date.getUTCMinutes();
 HEAP32[(tmPtr + 8) / 2 ** 2 >>> 0] = date.getUTCHours();
 HEAP32[(tmPtr + 12) / 2 ** 2 >>> 0] = date.getUTCDate();
 HEAP32[(tmPtr + 16) / 2 ** 2 >>> 0] = date.getUTCMonth();
 HEAP32[(tmPtr + 20) / 2 ** 2 >>> 0] = date.getUTCFullYear() - 1900;
 HEAP32[(tmPtr + 24) / 2 ** 2 >>> 0] = date.getUTCDay();
 var start = Date.UTC(date.getUTCFullYear(), 0, 1, 0, 0, 0, 0);
 var yday = (date.getTime() - start) / (1e3 * 60 * 60 * 24) | 0;
 HEAP32[(tmPtr + 28) / 2 ** 2 >>> 0] = yday;
}

function isLeapYear(year) {
 return year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0);
}

var MONTH_DAYS_LEAP_CUMULATIVE = [ 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335 ];

var MONTH_DAYS_REGULAR_CUMULATIVE = [ 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334 ];

function ydayFromDate(date) {
 var leap = isLeapYear(date.getFullYear());
 var monthDaysCumulative = leap ? MONTH_DAYS_LEAP_CUMULATIVE : MONTH_DAYS_REGULAR_CUMULATIVE;
 var yday = monthDaysCumulative[date.getMonth()] + date.getDate() - 1;
 return yday;
}

function __localtime_js(time, tmPtr) {
 time = Number(time);
 tmPtr = Number(tmPtr);
 var date = new Date(readI53FromI64(time) * 1e3);
 HEAP32[tmPtr / 2 ** 2 >>> 0] = date.getSeconds();
 HEAP32[(tmPtr + 4) / 2 ** 2 >>> 0] = date.getMinutes();
 HEAP32[(tmPtr + 8) / 2 ** 2 >>> 0] = date.getHours();
 HEAP32[(tmPtr + 12) / 2 ** 2 >>> 0] = date.getDate();
 HEAP32[(tmPtr + 16) / 2 ** 2 >>> 0] = date.getMonth();
 HEAP32[(tmPtr + 20) / 2 ** 2 >>> 0] = date.getFullYear() - 1900;
 HEAP32[(tmPtr + 24) / 2 ** 2 >>> 0] = date.getDay();
 var yday = ydayFromDate(date) | 0;
 HEAP32[(tmPtr + 28) / 2 ** 2 >>> 0] = yday;
 HEAP32[(tmPtr + 40) / 2 ** 2 >>> 0] = -(date.getTimezoneOffset() * 60);
 var start = new Date(date.getFullYear(), 0, 1);
 var summerOffset = new Date(date.getFullYear(), 6, 1).getTimezoneOffset();
 var winterOffset = start.getTimezoneOffset();
 var dst = (summerOffset != winterOffset && date.getTimezoneOffset() == Math.min(winterOffset, summerOffset)) | 0;
 HEAP32[(tmPtr + 32) / 2 ** 2 >>> 0] = dst;
}

function __mktime_js(tmPtr) {
 tmPtr = Number(tmPtr);
 var date = new Date(HEAP32[(tmPtr + 20) / 2 ** 2 >>> 0] + 1900, HEAP32[(tmPtr + 16) / 2 ** 2 >>> 0], HEAP32[(tmPtr + 12) / 2 ** 2 >>> 0], HEAP32[(tmPtr + 8) / 2 ** 2 >>> 0], HEAP32[(tmPtr + 4) / 2 ** 2 >>> 0], HEAP32[tmPtr / 2 ** 2 >>> 0], 0);
 var dst = HEAP32[(tmPtr + 32) / 2 ** 2 >>> 0];
 var guessedOffset = date.getTimezoneOffset();
 var start = new Date(date.getFullYear(), 0, 1);
 var summerOffset = new Date(date.getFullYear(), 6, 1).getTimezoneOffset();
 var winterOffset = start.getTimezoneOffset();
 var dstOffset = Math.min(winterOffset, summerOffset);
 if (dst < 0) {
  HEAP32[(tmPtr + 32) / 2 ** 2 >>> 0] = Number(summerOffset != winterOffset && dstOffset == guessedOffset);
 } else if (dst > 0 != (dstOffset == guessedOffset)) {
  var nonDstOffset = Math.max(winterOffset, summerOffset);
  var trueOffset = dst > 0 ? dstOffset : nonDstOffset;
  date.setTime(date.getTime() + (trueOffset - guessedOffset) * 6e4);
 }
 HEAP32[(tmPtr + 24) / 2 ** 2 >>> 0] = date.getDay();
 var yday = ydayFromDate(date) | 0;
 HEAP32[(tmPtr + 28) / 2 ** 2 >>> 0] = yday;
 HEAP32[tmPtr / 2 ** 2 >>> 0] = date.getSeconds();
 HEAP32[(tmPtr + 4) / 2 ** 2 >>> 0] = date.getMinutes();
 HEAP32[(tmPtr + 8) / 2 ** 2 >>> 0] = date.getHours();
 HEAP32[(tmPtr + 12) / 2 ** 2 >>> 0] = date.getDate();
 HEAP32[(tmPtr + 16) / 2 ** 2 >>> 0] = date.getMonth();
 HEAP32[(tmPtr + 20) / 2 ** 2 >>> 0] = date.getYear();
 return date.getTime() / 1e3 | 0;
}

function __mmap_js(len, prot, flags, fd, off, allocated, addr) {
 len = Number(len);
 off = Number(off);
 allocated = Number(allocated);
 addr = Number(addr);
 return -52;
}

function __munmap_js(addr, len, prot, flags, fd, offset) {
 addr = Number(addr);
 len = Number(len);
 offset = Number(offset);
}

function stringToNewUTF8(str) {
 var size = lengthBytesUTF8(str) + 1;
 var ret = _malloc(size);
 if (ret) stringToUTF8(str, ret, size);
 return ret;
}

function __tzset_js(timezone, daylight, tzname) {
 timezone = Number(timezone);
 daylight = Number(daylight);
 tzname = Number(tzname);
 var currentYear = new Date().getFullYear();
 var winter = new Date(currentYear, 0, 1);
 var summer = new Date(currentYear, 6, 1);
 var winterOffset = winter.getTimezoneOffset();
 var summerOffset = summer.getTimezoneOffset();
 var stdTimezoneOffset = Math.max(winterOffset, summerOffset);
 HEAPU64[timezone / 2 ** 3] = BigInt(stdTimezoneOffset * 60);
 HEAP32[daylight / 2 ** 2 >>> 0] = Number(winterOffset != summerOffset);
 function extractZone(date) {
  var match = date.toTimeString().match(/\(([A-Za-z ]+)\)$/);
  return match ? match[1] : "GMT";
 }
 var winterName = extractZone(winter);
 var summerName = extractZone(summer);
 var winterNamePtr = stringToNewUTF8(winterName);
 var summerNamePtr = stringToNewUTF8(summerName);
 if (summerOffset < winterOffset) {
  HEAPU64[tzname / 2 ** 3] = BigInt(winterNamePtr);
  HEAPU64[(tzname + 8) / 2 ** 3] = BigInt(summerNamePtr);
 } else {
  HEAPU64[tzname / 2 ** 3] = BigInt(summerNamePtr);
  HEAPU64[(tzname + 8) / 2 ** 3] = BigInt(winterNamePtr);
 }
}

function _abort() {
 abort("");
}

function _dlopen(filename, flags) {
 var ret = ((filename, flags) => {
  abort(dlopenMissingError);
 })(Number(filename), flags);
 return BigInt(ret);
}

function _emscripten_date_now() {
 return Date.now();
}

function getHeapMax() {
 return HEAPU8.length;
}

function _emscripten_get_heap_max() {
 var ret = (() => {
  return getHeapMax();
 })();
 return BigInt(ret);
}

var _emscripten_get_now;

if (ENVIRONMENT_IS_NODE) {
 global.performance = require("perf_hooks").performance;
}

_emscripten_get_now = () => performance.now();

function _emscripten_memcpy_big(dest, src, num) {
 dest = Number(dest);
 src = Number(src);
 num = Number(num);
 HEAPU8.copyWithin(fixPointer(dest), fixPointer(src), fixPointer(src) + fixPointer(num));
}

function abortOnCannotGrowMemory(requestedSize) {
 abort("OOM");
}

function _emscripten_resize_heap(requestedSize) {
 requestedSize = Number(requestedSize);
 var oldSize = HEAPU8.length;
 abortOnCannotGrowMemory(requestedSize);
}

var ENV = {};

function getExecutableName() {
 return thisProgram || "./this.program";
}

function getEnvStrings() {
 if (!getEnvStrings.strings) {
  var lang = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8";
  var env = {
   "USER": "web_user",
   "LOGNAME": "web_user",
   "PATH": "/",
   "PWD": "/",
   "HOME": "/home/web_user",
   "LANG": lang,
   "_": getExecutableName()
  };
  for (var x in ENV) {
   if (ENV[x] === undefined) delete env[x]; else env[x] = ENV[x];
  }
  var strings = [];
  for (var x in env) {
   strings.push(`${x}=${env[x]}`);
  }
  getEnvStrings.strings = strings;
 }
 return getEnvStrings.strings;
}

function stringToAscii(str, buffer) {
 buffer >>>= 0;
 for (var i = 0; i < str.length; ++i) {
  HEAP8[buffer++ / 2 ** 0 >>> 0] = str.charCodeAt(i);
 }
 HEAP8[buffer / 2 ** 0 >>> 0] = 0;
}

function _environ_get(__environ, environ_buf) {
 __environ = Number(__environ);
 environ_buf = Number(environ_buf);
 var bufSize = 0;
 getEnvStrings().forEach(function(string, i) {
  var ptr = environ_buf + bufSize;
  HEAPU64[(__environ + i * 8) / 2 ** 3] = BigInt(ptr);
  stringToAscii(string, ptr);
  bufSize += string.length + 1;
 });
 return 0;
}

function _environ_sizes_get(penviron_count, penviron_buf_size) {
 penviron_count = Number(penviron_count);
 penviron_buf_size = Number(penviron_buf_size);
 var strings = getEnvStrings();
 HEAPU64[penviron_count / 2 ** 3] = BigInt(strings.length);
 var bufSize = 0;
 strings.forEach(function(string) {
  bufSize += string.length + 1;
 });
 HEAPU64[penviron_buf_size / 2 ** 3] = BigInt(bufSize);
 return 0;
}

function _fd_close(fd) {
 return 52;
}

function _fd_read(fd, iov, iovcnt, pnum) {
 iov = Number(iov);
 iovcnt = Number(iovcnt);
 pnum = Number(pnum);
 return 52;
}

var MAX_INT53 = 9007199254740992;

var MIN_INT53 = -9007199254740992;

function bigintToI53Checked(num) {
 return num < MIN_INT53 || num > MAX_INT53 ? NaN : Number(num);
}

function _fd_seek(fd, offset, whence, newOffset) {
 newOffset = Number(newOffset);
 return 70;
}

var printCharBuffers = [ null, [], [] ];

function printChar(stream, curr) {
 var buffer = printCharBuffers[stream];
 if (curr === 0 || curr === 10) {
  (stream === 1 ? out : err)(UTF8ArrayToString(buffer, 0));
  buffer.length = 0;
 } else {
  buffer.push(curr);
 }
}

function flush_NO_FILESYSTEM() {
 if (printCharBuffers[1].length) printChar(1, 10);
 if (printCharBuffers[2].length) printChar(2, 10);
}

function _fd_write(fd, iov, iovcnt, pnum) {
 iov = Number(iov) >>> 0;
 iovcnt = Number(iovcnt) >>> 0;
 pnum = Number(pnum) >>> 0;
 var num = 0;
 for (var i = 0; i < iovcnt; i++) {
  var ptr = Number(HEAPU64[iov / 2 ** 3]);
  var len = Number(HEAPU64[(iov + 8) / 2 ** 3]);
  iov += 16;
  for (var j = 0; j < len; j++) {
   printChar(fd, HEAPU8[ptr + j >>> 0]);
  }
  num += len;
 }
 HEAPU64[pnum / 2 ** 3] = BigInt(num);
 return 0;
}

function _llvm_eh_typeid_for(type) {
 type = Number(type);
 return type;
}

function arraySum(array, index) {
 var sum = 0;
 for (var i = 0; i <= index; sum += array[i++]) {}
 return sum;
}

var MONTH_DAYS_LEAP = [ 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ];

var MONTH_DAYS_REGULAR = [ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ];

function addDays(date, days) {
 var newDate = new Date(date.getTime());
 while (days > 0) {
  var leap = isLeapYear(newDate.getFullYear());
  var currentMonth = newDate.getMonth();
  var daysInCurrentMonth = (leap ? MONTH_DAYS_LEAP : MONTH_DAYS_REGULAR)[currentMonth];
  if (days > daysInCurrentMonth - newDate.getDate()) {
   days -= daysInCurrentMonth - newDate.getDate() + 1;
   newDate.setDate(1);
   if (currentMonth < 11) {
    newDate.setMonth(currentMonth + 1);
   } else {
    newDate.setMonth(0);
    newDate.setFullYear(newDate.getFullYear() + 1);
   }
  } else {
   newDate.setDate(newDate.getDate() + days);
   return newDate;
  }
 }
 return newDate;
}

function intArrayFromString(stringy, dontAddNull, length) {
 var len = length > 0 ? length : lengthBytesUTF8(stringy) + 1;
 var u8array = new Array(len);
 var numBytesWritten = stringToUTF8Array(stringy, u8array, 0, u8array.length);
 if (dontAddNull) u8array.length = numBytesWritten;
 return u8array;
}

function writeArrayToMemory(array, buffer) {
 HEAP8.set(array, buffer >>> 0);
}

function _strftime(s, maxsize, format, tm) {
 var ret = ((s, maxsize, format, tm) => {
  var tm_zone = HEAP32[(tm + 48) / 2 ** 2 >>> 0];
  var date = {
   tm_sec: HEAP32[tm / 2 ** 2 >>> 0],
   tm_min: HEAP32[(tm + 4) / 2 ** 2 >>> 0],
   tm_hour: HEAP32[(tm + 8) / 2 ** 2 >>> 0],
   tm_mday: HEAP32[(tm + 12) / 2 ** 2 >>> 0],
   tm_mon: HEAP32[(tm + 16) / 2 ** 2 >>> 0],
   tm_year: HEAP32[(tm + 20) / 2 ** 2 >>> 0],
   tm_wday: HEAP32[(tm + 24) / 2 ** 2 >>> 0],
   tm_yday: HEAP32[(tm + 28) / 2 ** 2 >>> 0],
   tm_isdst: HEAP32[(tm + 32) / 2 ** 2 >>> 0],
   tm_gmtoff: HEAP32[(tm + 40) / 2 ** 2 >>> 0],
   tm_zone: tm_zone ? UTF8ToString(tm_zone) : ""
  };
  var pattern = UTF8ToString(format);
  var EXPANSION_RULES_1 = {
   "%c": "%a %b %d %H:%M:%S %Y",
   "%D": "%m/%d/%y",
   "%F": "%Y-%m-%d",
   "%h": "%b",
   "%r": "%I:%M:%S %p",
   "%R": "%H:%M",
   "%T": "%H:%M:%S",
   "%x": "%m/%d/%y",
   "%X": "%H:%M:%S",
   "%Ec": "%c",
   "%EC": "%C",
   "%Ex": "%m/%d/%y",
   "%EX": "%H:%M:%S",
   "%Ey": "%y",
   "%EY": "%Y",
   "%Od": "%d",
   "%Oe": "%e",
   "%OH": "%H",
   "%OI": "%I",
   "%Om": "%m",
   "%OM": "%M",
   "%OS": "%S",
   "%Ou": "%u",
   "%OU": "%U",
   "%OV": "%V",
   "%Ow": "%w",
   "%OW": "%W",
   "%Oy": "%y"
  };
  for (var rule in EXPANSION_RULES_1) {
   pattern = pattern.replace(new RegExp(rule, "g"), EXPANSION_RULES_1[rule]);
  }
  var WEEKDAYS = [ "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday" ];
  var MONTHS = [ "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December" ];
  function leadingSomething(value, digits, character) {
   var str = typeof value == "number" ? value.toString() : value || "";
   while (str.length < digits) {
    str = character[0] + str;
   }
   return str;
  }
  function leadingNulls(value, digits) {
   return leadingSomething(value, digits, "0");
  }
  function compareByDay(date1, date2) {
   function sgn(value) {
    return value < 0 ? -1 : value > 0 ? 1 : 0;
   }
   var compare;
   if ((compare = sgn(date1.getFullYear() - date2.getFullYear())) === 0) {
    if ((compare = sgn(date1.getMonth() - date2.getMonth())) === 0) {
     compare = sgn(date1.getDate() - date2.getDate());
    }
   }
   return compare;
  }
  function getFirstWeekStartDate(janFourth) {
   switch (janFourth.getDay()) {
   case 0:
    return new Date(janFourth.getFullYear() - 1, 11, 29);

   case 1:
    return janFourth;

   case 2:
    return new Date(janFourth.getFullYear(), 0, 3);

   case 3:
    return new Date(janFourth.getFullYear(), 0, 2);

   case 4:
    return new Date(janFourth.getFullYear(), 0, 1);

   case 5:
    return new Date(janFourth.getFullYear() - 1, 11, 31);

   case 6:
    return new Date(janFourth.getFullYear() - 1, 11, 30);
   }
  }
  function getWeekBasedYear(date) {
   var thisDate = addDays(new Date(date.tm_year + 1900, 0, 1), date.tm_yday);
   var janFourthThisYear = new Date(thisDate.getFullYear(), 0, 4);
   var janFourthNextYear = new Date(thisDate.getFullYear() + 1, 0, 4);
   var firstWeekStartThisYear = getFirstWeekStartDate(janFourthThisYear);
   var firstWeekStartNextYear = getFirstWeekStartDate(janFourthNextYear);
   if (compareByDay(firstWeekStartThisYear, thisDate) <= 0) {
    if (compareByDay(firstWeekStartNextYear, thisDate) <= 0) {
     return thisDate.getFullYear() + 1;
    }
    return thisDate.getFullYear();
   }
   return thisDate.getFullYear() - 1;
  }
  var EXPANSION_RULES_2 = {
   "%a": function(date) {
    return WEEKDAYS[date.tm_wday].substring(0, 3);
   },
   "%A": function(date) {
    return WEEKDAYS[date.tm_wday];
   },
   "%b": function(date) {
    return MONTHS[date.tm_mon].substring(0, 3);
   },
   "%B": function(date) {
    return MONTHS[date.tm_mon];
   },
   "%C": function(date) {
    var year = date.tm_year + 1900;
    return leadingNulls(year / 100 | 0, 2);
   },
   "%d": function(date) {
    return leadingNulls(date.tm_mday, 2);
   },
   "%e": function(date) {
    return leadingSomething(date.tm_mday, 2, " ");
   },
   "%g": function(date) {
    return getWeekBasedYear(date).toString().substring(2);
   },
   "%G": function(date) {
    return getWeekBasedYear(date);
   },
   "%H": function(date) {
    return leadingNulls(date.tm_hour, 2);
   },
   "%I": function(date) {
    var twelveHour = date.tm_hour;
    if (twelveHour == 0) twelveHour = 12; else if (twelveHour > 12) twelveHour -= 12;
    return leadingNulls(twelveHour, 2);
   },
   "%j": function(date) {
    return leadingNulls(date.tm_mday + arraySum(isLeapYear(date.tm_year + 1900) ? MONTH_DAYS_LEAP : MONTH_DAYS_REGULAR, date.tm_mon - 1), 3);
   },
   "%m": function(date) {
    return leadingNulls(date.tm_mon + 1, 2);
   },
   "%M": function(date) {
    return leadingNulls(date.tm_min, 2);
   },
   "%n": function() {
    return "\n";
   },
   "%p": function(date) {
    if (date.tm_hour >= 0 && date.tm_hour < 12) {
     return "AM";
    }
    return "PM";
   },
   "%S": function(date) {
    return leadingNulls(date.tm_sec, 2);
   },
   "%t": function() {
    return "\t";
   },
   "%u": function(date) {
    return date.tm_wday || 7;
   },
   "%U": function(date) {
    var days = date.tm_yday + 7 - date.tm_wday;
    return leadingNulls(Math.floor(days / 7), 2);
   },
   "%V": function(date) {
    var val = Math.floor((date.tm_yday + 7 - (date.tm_wday + 6) % 7) / 7);
    if ((date.tm_wday + 371 - date.tm_yday - 2) % 7 <= 2) {
     val++;
    }
    if (!val) {
     val = 52;
     var dec31 = (date.tm_wday + 7 - date.tm_yday - 1) % 7;
     if (dec31 == 4 || dec31 == 5 && isLeapYear(date.tm_year % 400 - 1)) {
      val++;
     }
    } else if (val == 53) {
     var jan1 = (date.tm_wday + 371 - date.tm_yday) % 7;
     if (jan1 != 4 && (jan1 != 3 || !isLeapYear(date.tm_year))) val = 1;
    }
    return leadingNulls(val, 2);
   },
   "%w": function(date) {
    return date.tm_wday;
   },
   "%W": function(date) {
    var days = date.tm_yday + 7 - (date.tm_wday + 6) % 7;
    return leadingNulls(Math.floor(days / 7), 2);
   },
   "%y": function(date) {
    return (date.tm_year + 1900).toString().substring(2);
   },
   "%Y": function(date) {
    return date.tm_year + 1900;
   },
   "%z": function(date) {
    var off = date.tm_gmtoff;
    var ahead = off >= 0;
    off = Math.abs(off) / 60;
    off = off / 60 * 100 + off % 60;
    return (ahead ? "+" : "-") + String("0000" + off).slice(-4);
   },
   "%Z": function(date) {
    return date.tm_zone;
   },
   "%%": function() {
    return "%";
   }
  };
  pattern = pattern.replace(/%%/g, "\0\0");
  for (var rule in EXPANSION_RULES_2) {
   if (pattern.includes(rule)) {
    pattern = pattern.replace(new RegExp(rule, "g"), EXPANSION_RULES_2[rule](date));
   }
  }
  pattern = pattern.replace(/\0\0/g, "%");
  var bytes = intArrayFromString(pattern, false);
  if (bytes.length > maxsize) {
   return 0;
  }
  writeArrayToMemory(bytes, s);
  return bytes.length - 1;
 })(Number(s), Number(maxsize), Number(format), Number(tm));
 return BigInt(ret);
}

function _strftime_l(s, maxsize, format, tm, loc) {
 var ret = ((s, maxsize, format, tm, loc) => {
  return _strftime(s, maxsize, format, tm);
 })(Number(s), Number(maxsize), Number(format), Number(tm), Number(loc));
 return BigInt(ret);
}

var wasmTableMirror = [];

function getWasmTableEntry(funcPtr) {
 funcPtr = Number(funcPtr);
 var func = wasmTableMirror[funcPtr];
 if (!func) {
  if (funcPtr >= wasmTableMirror.length) wasmTableMirror.length = funcPtr + 1;
  wasmTableMirror[funcPtr] = func = wasmTable.get(funcPtr);
 }
 return func;
}

embind_init_charCodes();

BindingError = Module["BindingError"] = extendError(Error, "BindingError");

InternalError = Module["InternalError"] = extendError(Error, "InternalError");

init_emval();

var wasmImports = {
 "__cxa_begin_catch": ___cxa_begin_catch,
 "__cxa_end_catch": ___cxa_end_catch,
 "__cxa_find_matching_catch_2": ___cxa_find_matching_catch_2,
 "__cxa_find_matching_catch_3": ___cxa_find_matching_catch_3,
 "__cxa_find_matching_catch_4": ___cxa_find_matching_catch_4,
 "__cxa_find_matching_catch_5": ___cxa_find_matching_catch_5,
 "__cxa_rethrow": ___cxa_rethrow,
 "__cxa_throw": ___cxa_throw,
 "__cxa_uncaught_exceptions": ___cxa_uncaught_exceptions,
 "__dlsym": ___dlsym,
 "__resumeException": ___resumeException,
 "__syscall_fcntl64": ___syscall_fcntl64,
 "__syscall_fstat64": ___syscall_fstat64,
 "__syscall_getcwd": ___syscall_getcwd,
 "__syscall_getdents64": ___syscall_getdents64,
 "__syscall_ioctl": ___syscall_ioctl,
 "__syscall_lstat64": ___syscall_lstat64,
 "__syscall_mkdirat": ___syscall_mkdirat,
 "__syscall_newfstatat": ___syscall_newfstatat,
 "__syscall_openat": ___syscall_openat,
 "__syscall_readlinkat": ___syscall_readlinkat,
 "__syscall_rmdir": ___syscall_rmdir,
 "__syscall_stat64": ___syscall_stat64,
 "__syscall_unlinkat": ___syscall_unlinkat,
 "_embind_register_bigint": __embind_register_bigint,
 "_embind_register_bool": __embind_register_bool,
 "_embind_register_emval": __embind_register_emval,
 "_embind_register_float": __embind_register_float,
 "_embind_register_integer": __embind_register_integer,
 "_embind_register_memory_view": __embind_register_memory_view,
 "_embind_register_std_string": __embind_register_std_string,
 "_embind_register_std_wstring": __embind_register_std_wstring,
 "_embind_register_void": __embind_register_void,
 "_emscripten_get_now_is_monotonic": __emscripten_get_now_is_monotonic,
 "_emval_as": __emval_as,
 "_emval_call_method": __emval_call_method,
 "_emval_call_void_method": __emval_call_void_method,
 "_emval_decref": __emval_decref,
 "_emval_equals": __emval_equals,
 "_emval_get_global": __emval_get_global,
 "_emval_get_method_caller": __emval_get_method_caller,
 "_emval_get_property": __emval_get_property,
 "_emval_incref": __emval_incref,
 "_emval_new": __emval_new,
 "_emval_new_array": __emval_new_array,
 "_emval_new_array_from_memory_view": __emval_new_array_from_memory_view,
 "_emval_new_cstring": __emval_new_cstring,
 "_emval_new_object": __emval_new_object,
 "_emval_run_destructors": __emval_run_destructors,
 "_emval_set_property": __emval_set_property,
 "_emval_take_value": __emval_take_value,
 "_gmtime_js": __gmtime_js,
 "_localtime_js": __localtime_js,
 "_mktime_js": __mktime_js,
 "_mmap_js": __mmap_js,
 "_munmap_js": __munmap_js,
 "_tzset_js": __tzset_js,
 "abort": _abort,
 "dlopen": _dlopen,
 "emscripten_date_now": _emscripten_date_now,
 "emscripten_get_heap_max": _emscripten_get_heap_max,
 "emscripten_get_now": _emscripten_get_now,
 "emscripten_memcpy_big": _emscripten_memcpy_big,
 "emscripten_resize_heap": _emscripten_resize_heap,
 "environ_get": _environ_get,
 "environ_sizes_get": _environ_sizes_get,
 "fd_close": _fd_close,
 "fd_read": _fd_read,
 "fd_seek": _fd_seek,
 "fd_write": _fd_write,
 "invoke_dj": invoke_dj,
 "invoke_djj": invoke_djj,
 "invoke_djjj": invoke_djjj,
 "invoke_djjjjj": invoke_djjjjj,
 "invoke_fffffff": invoke_fffffff,
 "invoke_fj": invoke_fj,
 "invoke_fjj": invoke_fjj,
 "invoke_fjjj": invoke_fjjj,
 "invoke_fjjjj": invoke_fjjjj,
 "invoke_i": invoke_i,
 "invoke_iii": invoke_iii,
 "invoke_iiiij": invoke_iiiij,
 "invoke_iij": invoke_iij,
 "invoke_iijjj": invoke_iijjj,
 "invoke_iijjjj": invoke_iijjjj,
 "invoke_ij": invoke_ij,
 "invoke_iji": invoke_iji,
 "invoke_ijii": invoke_ijii,
 "invoke_ijiiji": invoke_ijiiji,
 "invoke_ijij": invoke_ijij,
 "invoke_ijijj": invoke_ijijj,
 "invoke_ijijjjj": invoke_ijijjjj,
 "invoke_ijijjjjjj": invoke_ijijjjjjj,
 "invoke_ijj": invoke_ijj,
 "invoke_ijji": invoke_ijji,
 "invoke_ijjii": invoke_ijjii,
 "invoke_ijjij": invoke_ijjij,
 "invoke_ijjijij": invoke_ijjijij,
 "invoke_ijjijijjjjjj": invoke_ijjijijjjjjj,
 "invoke_ijjijjj": invoke_ijjijjj,
 "invoke_ijjj": invoke_ijjj,
 "invoke_ijjjfi": invoke_ijjjfi,
 "invoke_ijjji": invoke_ijjji,
 "invoke_ijjjij": invoke_ijjjij,
 "invoke_ijjjijij": invoke_ijjjijij,
 "invoke_ijjjijj": invoke_ijjjijj,
 "invoke_ijjjijjjji": invoke_ijjjijjjji,
 "invoke_ijjjijjjjijjjj": invoke_ijjjijjjjijjjj,
 "invoke_ijjjj": invoke_ijjjj,
 "invoke_ijjjji": invoke_ijjjji,
 "invoke_ijjjjij": invoke_ijjjjij,
 "invoke_ijjjjj": invoke_ijjjjj,
 "invoke_ijjjjjij": invoke_ijjjjjij,
 "invoke_ijjjjjj": invoke_ijjjjjj,
 "invoke_ijjjjjjj": invoke_ijjjjjjj,
 "invoke_ijjjjjjjii": invoke_ijjjjjjjii,
 "invoke_ijjjjjjjjjj": invoke_ijjjjjjjjjj,
 "invoke_ijjjjjjjjjjfj": invoke_ijjjjjjjjjjfj,
 "invoke_ijjjjjjjjjjj": invoke_ijjjjjjjjjjj,
 "invoke_j": invoke_j,
 "invoke_jfi": invoke_jfi,
 "invoke_ji": invoke_ji,
 "invoke_jii": invoke_jii,
 "invoke_jij": invoke_jij,
 "invoke_jiji": invoke_jiji,
 "invoke_jijiii": invoke_jijiii,
 "invoke_jijj": invoke_jijj,
 "invoke_jj": invoke_jj,
 "invoke_jjd": invoke_jjd,
 "invoke_jjf": invoke_jjf,
 "invoke_jji": invoke_jji,
 "invoke_jjii": invoke_jjii,
 "invoke_jjiiijjjjjjj": invoke_jjiiijjjjjjj,
 "invoke_jjiij": invoke_jjiij,
 "invoke_jjij": invoke_jjij,
 "invoke_jjijj": invoke_jjijj,
 "invoke_jjijjiii": invoke_jjijjiii,
 "invoke_jjijjiiii": invoke_jjijjiiii,
 "invoke_jjijjj": invoke_jjijjj,
 "invoke_jjijjjij": invoke_jjijjjij,
 "invoke_jjj": invoke_jjj,
 "invoke_jjjf": invoke_jjjf,
 "invoke_jjji": invoke_jjji,
 "invoke_jjjii": invoke_jjjii,
 "invoke_jjjiiiiiijjjjjjfj": invoke_jjjiiiiiijjjjjjfj,
 "invoke_jjjiiiji": invoke_jjjiiiji,
 "invoke_jjjij": invoke_jjjij,
 "invoke_jjjijij": invoke_jjjijij,
 "invoke_jjjijj": invoke_jjjijj,
 "invoke_jjjijjj": invoke_jjjijjj,
 "invoke_jjjj": invoke_jjjj,
 "invoke_jjjji": invoke_jjjji,
 "invoke_jjjjid": invoke_jjjjid,
 "invoke_jjjjii": invoke_jjjjii,
 "invoke_jjjjiiiiiij": invoke_jjjjiiiiiij,
 "invoke_jjjjiiiiiijjjjjjjfji": invoke_jjjjiiiiiijjjjjjjfji,
 "invoke_jjjjiiiiiijjjjjjjjfj": invoke_jjjjiiiiiijjjjjjjjfj,
 "invoke_jjjjiiiiijj": invoke_jjjjiiiiijj,
 "invoke_jjjjiiiij": invoke_jjjjiiiij,
 "invoke_jjjjij": invoke_jjjjij,
 "invoke_jjjjijij": invoke_jjjjijij,
 "invoke_jjjjijj": invoke_jjjjijj,
 "invoke_jjjjj": invoke_jjjjj,
 "invoke_jjjjji": invoke_jjjjji,
 "invoke_jjjjjij": invoke_jjjjjij,
 "invoke_jjjjjj": invoke_jjjjjj,
 "invoke_jjjjjji": invoke_jjjjjji,
 "invoke_jjjjjjj": invoke_jjjjjjj,
 "invoke_jjjjjjji": invoke_jjjjjjji,
 "invoke_jjjjjjjj": invoke_jjjjjjjj,
 "invoke_jjjjjjjji": invoke_jjjjjjjji,
 "invoke_jjjjjjjjijj": invoke_jjjjjjjjijj,
 "invoke_jjjjjjjjj": invoke_jjjjjjjjj,
 "invoke_jjjjjjjjjj": invoke_jjjjjjjjjj,
 "invoke_jjjjjjjjjji": invoke_jjjjjjjjjji,
 "invoke_jjjjjjjjjjj": invoke_jjjjjjjjjjj,
 "invoke_jjjjjjjjjjjj": invoke_jjjjjjjjjjjj,
 "invoke_v": invoke_v,
 "invoke_vfjji": invoke_vfjji,
 "invoke_vi": invoke_vi,
 "invoke_viiiiiiffjfjjjjj": invoke_viiiiiiffjfjjjjj,
 "invoke_viiiiiiffjifjjjjj": invoke_viiiiiiffjifjjjjj,
 "invoke_viiiiiiffjjjjjj": invoke_viiiiiiffjjjjjj,
 "invoke_viiijjijjifjji": invoke_viiijjijjifjji,
 "invoke_viijjjdjjdj": invoke_viijjjdjjdj,
 "invoke_viijjjjj": invoke_viijjjjj,
 "invoke_viijjjjjjjj": invoke_viijjjjjjjj,
 "invoke_vij": invoke_vij,
 "invoke_vijij": invoke_vijij,
 "invoke_vijj": invoke_vijj,
 "invoke_vijjji": invoke_vijjji,
 "invoke_vijjjjjjjjj": invoke_vijjjjjjjjj,
 "invoke_vj": invoke_vj,
 "invoke_vjV": invoke_vjV,
 "invoke_vjdi": invoke_vjdi,
 "invoke_vjf": invoke_vjf,
 "invoke_vjffjij": invoke_vjffjij,
 "invoke_vji": invoke_vji,
 "invoke_vjiff": invoke_vjiff,
 "invoke_vjii": invoke_vjii,
 "invoke_vjiii": invoke_vjiii,
 "invoke_vjiij": invoke_vjiij,
 "invoke_vjiijj": invoke_vjiijj,
 "invoke_vjij": invoke_vjij,
 "invoke_vjiji": invoke_vjiji,
 "invoke_vjijj": invoke_vjijj,
 "invoke_vjijjj": invoke_vjijjj,
 "invoke_vjijjji": invoke_vjijjji,
 "invoke_vjijjjj": invoke_vjijjjj,
 "invoke_vjijjjjjj": invoke_vjijjjjjj,
 "invoke_vjijjjjjjjjji": invoke_vjijjjjjjjjji,
 "invoke_vjj": invoke_vjj,
 "invoke_vjjd": invoke_vjjd,
 "invoke_vjjf": invoke_vjjf,
 "invoke_vjjfijfijjjij": invoke_vjjfijfijjjij,
 "invoke_vjjfj": invoke_vjjfj,
 "invoke_vjjfjjj": invoke_vjjfjjj,
 "invoke_vjji": invoke_vjji,
 "invoke_vjjii": invoke_vjjii,
 "invoke_vjjiiii": invoke_vjjiiii,
 "invoke_vjjiiiijjjjj": invoke_vjjiiiijjjjj,
 "invoke_vjjiiij": invoke_vjjiiij,
 "invoke_vjjiijjj": invoke_vjjiijjj,
 "invoke_vjjij": invoke_vjjij,
 "invoke_vjjiji": invoke_vjjiji,
 "invoke_vjjijijj": invoke_vjjijijj,
 "invoke_vjjijj": invoke_vjjijj,
 "invoke_vjjijjj": invoke_vjjijjj,
 "invoke_vjjj": invoke_vjjj,
 "invoke_vjjjd": invoke_vjjjd,
 "invoke_vjjjf": invoke_vjjjf,
 "invoke_vjjjfffijj": invoke_vjjjfffijj,
 "invoke_vjjjfij": invoke_vjjjfij,
 "invoke_vjjji": invoke_vjjji,
 "invoke_vjjjiff": invoke_vjjjiff,
 "invoke_vjjjii": invoke_vjjjii,
 "invoke_vjjjiii": invoke_vjjjiii,
 "invoke_vjjjiiiiiiii": invoke_vjjjiiiiiiii,
 "invoke_vjjjiiiijjij": invoke_vjjjiiiijjij,
 "invoke_vjjjiij": invoke_vjjjiij,
 "invoke_vjjjiijjjjjii": invoke_vjjjiijjjjjii,
 "invoke_vjjjij": invoke_vjjjij,
 "invoke_vjjjiji": invoke_vjjjiji,
 "invoke_vjjjijj": invoke_vjjjijj,
 "invoke_vjjjijji": invoke_vjjjijji,
 "invoke_vjjjijjj": invoke_vjjjijjj,
 "invoke_vjjjijjjijiijjji": invoke_vjjjijjjijiijjji,
 "invoke_vjjjj": invoke_vjjjj,
 "invoke_vjjjjfjjj": invoke_vjjjjfjjj,
 "invoke_vjjjji": invoke_vjjjji,
 "invoke_vjjjjiff": invoke_vjjjjiff,
 "invoke_vjjjjii": invoke_vjjjjii,
 "invoke_vjjjjiii": invoke_vjjjjiii,
 "invoke_vjjjjiiii": invoke_vjjjjiiii,
 "invoke_vjjjjiij": invoke_vjjjjiij,
 "invoke_vjjjjiijjjjjjjjii": invoke_vjjjjiijjjjjjjjii,
 "invoke_vjjjjij": invoke_vjjjjij,
 "invoke_vjjjjijj": invoke_vjjjjijj,
 "invoke_vjjjjijjjjj": invoke_vjjjjijjjjj,
 "invoke_vjjjjijjjjjjj": invoke_vjjjjijjjjjjj,
 "invoke_vjjjjijjjjjjjj": invoke_vjjjjijjjjjjjj,
 "invoke_vjjjjj": invoke_vjjjjj,
 "invoke_vjjjjjfiijjjj": invoke_vjjjjjfiijjjj,
 "invoke_vjjjjji": invoke_vjjjjji,
 "invoke_vjjjjjii": invoke_vjjjjjii,
 "invoke_vjjjjjiiij": invoke_vjjjjjiiij,
 "invoke_vjjjjjiijjjjjj": invoke_vjjjjjiijjjjjj,
 "invoke_vjjjjjij": invoke_vjjjjjij,
 "invoke_vjjjjjijj": invoke_vjjjjjijj,
 "invoke_vjjjjjijji": invoke_vjjjjjijji,
 "invoke_vjjjjjijjjj": invoke_vjjjjjijjjj,
 "invoke_vjjjjjijjjjj": invoke_vjjjjjijjjjj,
 "invoke_vjjjjjj": invoke_vjjjjjj,
 "invoke_vjjjjjjfffifijjjj": invoke_vjjjjjjfffifijjjj,
 "invoke_vjjjjjjfffifijjjjjj": invoke_vjjjjjjfffifijjjjjj,
 "invoke_vjjjjjjffjifijjjjj": invoke_vjjjjjjffjifijjjjj,
 "invoke_vjjjjjjfjj": invoke_vjjjjjjfjj,
 "invoke_vjjjjjji": invoke_vjjjjjji,
 "invoke_vjjjjjjiiijjj": invoke_vjjjjjjiiijjj,
 "invoke_vjjjjjjij": invoke_vjjjjjjij,
 "invoke_vjjjjjjijjj": invoke_vjjjjjjijjj,
 "invoke_vjjjjjjijjjiiiiiiiijj": invoke_vjjjjjjijjjiiiiiiiijj,
 "invoke_vjjjjjjijjjj": invoke_vjjjjjjijjjj,
 "invoke_vjjjjjjj": invoke_vjjjjjjj,
 "invoke_vjjjjjjji": invoke_vjjjjjjji,
 "invoke_vjjjjjjjifjjVj": invoke_vjjjjjjjifjjVj,
 "invoke_vjjjjjjjiifijj": invoke_vjjjjjjjiifijj,
 "invoke_vjjjjjjjiiiijj": invoke_vjjjjjjjiiiijj,
 "invoke_vjjjjjjjij": invoke_vjjjjjjjij,
 "invoke_vjjjjjjjijj": invoke_vjjjjjjjijj,
 "invoke_vjjjjjjjj": invoke_vjjjjjjjj,
 "invoke_vjjjjjjjjfffjifijjjjj": invoke_vjjjjjjjjfffjifijjjjj,
 "invoke_vjjjjjjjjfffjifjjjjj": invoke_vjjjjjjjjfffjifjjjjj,
 "invoke_vjjjjjjjji": invoke_vjjjjjjjji,
 "invoke_vjjjjjjjjj": invoke_vjjjjjjjjj,
 "invoke_vjjjjjjjjjdj": invoke_vjjjjjjjjjdj,
 "invoke_vjjjjjjjjjfj": invoke_vjjjjjjjjjfj,
 "invoke_vjjjjjjjjji": invoke_vjjjjjjjjji,
 "invoke_vjjjjjjjjjiijj": invoke_vjjjjjjjjjiijj,
 "invoke_vjjjjjjjjjij": invoke_vjjjjjjjjjij,
 "invoke_vjjjjjjjjjj": invoke_vjjjjjjjjjj,
 "invoke_vjjjjjjjjjji": invoke_vjjjjjjjjjji,
 "invoke_vjjjjjjjjjjii": invoke_vjjjjjjjjjjii,
 "invoke_vjjjjjjjjjjiijj": invoke_vjjjjjjjjjjiijj,
 "invoke_vjjjjjjjjjjj": invoke_vjjjjjjjjjjj,
 "invoke_vjjjjjjjjjjjijiijii": invoke_vjjjjjjjjjjjijiijii,
 "invoke_vjjjjjjjjjjjjiiiiiijj": invoke_vjjjjjjjjjjjjiiiiiijj,
 "invoke_vjjjjjjjjjjjjijiijii": invoke_vjjjjjjjjjjjjijiijii,
 "invoke_vjjjjjjjjjjjjj": invoke_vjjjjjjjjjjjjj,
 "invoke_vjjjjjjjjjjjjjj": invoke_vjjjjjjjjjjjjjj,
 "invoke_vjjjjjjjjjjjjjjjjjjj": invoke_vjjjjjjjjjjjjjjjjjjj,
 "invoke_vjjjjjjjjjjjjjjjjjjjj": invoke_vjjjjjjjjjjjjjjjjjjjj,
 "invoke_vjjjjjjjjjjjjjjjjjjjjjjjjj": invoke_vjjjjjjjjjjjjjjjjjjjjjjjjj,
 "llvm_eh_typeid_for": _llvm_eh_typeid_for,
 "strftime": _strftime,
 "strftime_l": _strftime_l
};

var asm = createWasm();

var ___wasm_call_ctors = function() {
 return (___wasm_call_ctors = Module["asm"]["__wasm_call_ctors"]).apply(null, arguments);
};

var _OrtInit = Module["_OrtInit"] = function() {
 return (_OrtInit = Module["_OrtInit"] = Module["asm"]["OrtInit"]).apply(null, arguments);
};

var _OrtCreateSessionOptions = Module["_OrtCreateSessionOptions"] = function() {
 return (_OrtCreateSessionOptions = Module["_OrtCreateSessionOptions"] = Module["asm"]["OrtCreateSessionOptions"]).apply(null, arguments);
};

var _OrtAppendExecutionProvider = Module["_OrtAppendExecutionProvider"] = function() {
 return (_OrtAppendExecutionProvider = Module["_OrtAppendExecutionProvider"] = Module["asm"]["OrtAppendExecutionProvider"]).apply(null, arguments);
};

var _OrtAddSessionConfigEntry = Module["_OrtAddSessionConfigEntry"] = function() {
 return (_OrtAddSessionConfigEntry = Module["_OrtAddSessionConfigEntry"] = Module["asm"]["OrtAddSessionConfigEntry"]).apply(null, arguments);
};

var _OrtReleaseSessionOptions = Module["_OrtReleaseSessionOptions"] = function() {
 return (_OrtReleaseSessionOptions = Module["_OrtReleaseSessionOptions"] = Module["asm"]["OrtReleaseSessionOptions"]).apply(null, arguments);
};

var _OrtCreateSession = Module["_OrtCreateSession"] = function() {
 return (_OrtCreateSession = Module["_OrtCreateSession"] = Module["asm"]["OrtCreateSession"]).apply(null, arguments);
};

var _OrtReleaseSession = Module["_OrtReleaseSession"] = function() {
 return (_OrtReleaseSession = Module["_OrtReleaseSession"] = Module["asm"]["OrtReleaseSession"]).apply(null, arguments);
};

var _OrtGetInputCount = Module["_OrtGetInputCount"] = function() {
 return (_OrtGetInputCount = Module["_OrtGetInputCount"] = Module["asm"]["OrtGetInputCount"]).apply(null, arguments);
};

var _OrtGetOutputCount = Module["_OrtGetOutputCount"] = function() {
 return (_OrtGetOutputCount = Module["_OrtGetOutputCount"] = Module["asm"]["OrtGetOutputCount"]).apply(null, arguments);
};

var _OrtGetInputName = Module["_OrtGetInputName"] = function() {
 return (_OrtGetInputName = Module["_OrtGetInputName"] = Module["asm"]["OrtGetInputName"]).apply(null, arguments);
};

var _OrtGetOutputName = Module["_OrtGetOutputName"] = function() {
 return (_OrtGetOutputName = Module["_OrtGetOutputName"] = Module["asm"]["OrtGetOutputName"]).apply(null, arguments);
};

var _OrtFree = Module["_OrtFree"] = function() {
 return (_OrtFree = Module["_OrtFree"] = Module["asm"]["OrtFree"]).apply(null, arguments);
};

var _OrtCreateTensor = Module["_OrtCreateTensor"] = function() {
 return (_OrtCreateTensor = Module["_OrtCreateTensor"] = Module["asm"]["OrtCreateTensor"]).apply(null, arguments);
};

var _OrtGetTensorData = Module["_OrtGetTensorData"] = function() {
 return (_OrtGetTensorData = Module["_OrtGetTensorData"] = Module["asm"]["OrtGetTensorData"]).apply(null, arguments);
};

var _OrtReleaseTensor = Module["_OrtReleaseTensor"] = function() {
 return (_OrtReleaseTensor = Module["_OrtReleaseTensor"] = Module["asm"]["OrtReleaseTensor"]).apply(null, arguments);
};

var _OrtCreateRunOptions = Module["_OrtCreateRunOptions"] = function() {
 return (_OrtCreateRunOptions = Module["_OrtCreateRunOptions"] = Module["asm"]["OrtCreateRunOptions"]).apply(null, arguments);
};

var _OrtAddRunConfigEntry = Module["_OrtAddRunConfigEntry"] = function() {
 return (_OrtAddRunConfigEntry = Module["_OrtAddRunConfigEntry"] = Module["asm"]["OrtAddRunConfigEntry"]).apply(null, arguments);
};

var _OrtReleaseRunOptions = Module["_OrtReleaseRunOptions"] = function() {
 return (_OrtReleaseRunOptions = Module["_OrtReleaseRunOptions"] = Module["asm"]["OrtReleaseRunOptions"]).apply(null, arguments);
};

var _OrtRun = Module["_OrtRun"] = function() {
 return (_OrtRun = Module["_OrtRun"] = Module["asm"]["OrtRun"]).apply(null, arguments);
};

var _OrtEndProfiling = Module["_OrtEndProfiling"] = function() {
 return (_OrtEndProfiling = Module["_OrtEndProfiling"] = Module["asm"]["OrtEndProfiling"]).apply(null, arguments);
};

var ___cxa_free_exception = function() {
 return (___cxa_free_exception = Module["asm"]["__cxa_free_exception"]).apply(null, arguments);
};

var ___errno_location = function() {
 return (___errno_location = Module["asm"]["__errno_location"]).apply(null, arguments);
};

var _malloc = Module["_malloc"] = function() {
 return (_malloc = Module["_malloc"] = Module["asm"]["malloc"]).apply(null, arguments);
};

var _free = Module["_free"] = function() {
 return (_free = Module["_free"] = Module["asm"]["free"]).apply(null, arguments);
};

var ___getTypeName = function() {
 return (___getTypeName = Module["asm"]["__getTypeName"]).apply(null, arguments);
};

var __embind_initialize_bindings = Module["__embind_initialize_bindings"] = function() {
 return (__embind_initialize_bindings = Module["__embind_initialize_bindings"] = Module["asm"]["_embind_initialize_bindings"]).apply(null, arguments);
};

var _setThrew = function() {
 return (_setThrew = Module["asm"]["setThrew"]).apply(null, arguments);
};

var setTempRet0 = function() {
 return (setTempRet0 = Module["asm"]["setTempRet0"]).apply(null, arguments);
};

var stackSave = function() {
 return (stackSave = Module["asm"]["stackSave"]).apply(null, arguments);
};

var stackRestore = function() {
 return (stackRestore = Module["asm"]["stackRestore"]).apply(null, arguments);
};

var stackAlloc = function() {
 return (stackAlloc = Module["asm"]["stackAlloc"]).apply(null, arguments);
};

var ___cxa_decrement_exception_refcount = function() {
 return (___cxa_decrement_exception_refcount = Module["asm"]["__cxa_decrement_exception_refcount"]).apply(null, arguments);
};

var ___cxa_increment_exception_refcount = function() {
 return (___cxa_increment_exception_refcount = Module["asm"]["__cxa_increment_exception_refcount"]).apply(null, arguments);
};

var ___cxa_can_catch = function() {
 return (___cxa_can_catch = Module["asm"]["__cxa_can_catch"]).apply(null, arguments);
};

var ___cxa_is_pointer_type = function() {
 return (___cxa_is_pointer_type = Module["asm"]["__cxa_is_pointer_type"]).apply(null, arguments);
};

function invoke_ij(index, a1) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vj(index, a1) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjj(index, a1, a2) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjj(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_iji(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jji(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jj(index, a1) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjjji(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vji(index, a1, a2) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_v(index) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))();
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjijj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjiij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjji(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijj(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_j(index) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))();
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjijij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_iij(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijji(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjii(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjij(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ijjj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjijjjij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjii(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_fj(index, a1) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_dj(index, a1) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjii(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjijjiiii(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjii(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijjji(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vijjji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjji(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ji(index, a1) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjijjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjij(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_i(index) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))();
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiji(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjji(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjij(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjij(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjijj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijij(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_iiiij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjijj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vij(index, a1, a2) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjiij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjji(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjji(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjd(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjiiiij(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjijjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjijji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjV(index, a1, a2) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjijjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vijj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjii(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjffjij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjijij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjijj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjfjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viijjjjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjijjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viijjjdjjdj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjd(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjiiii(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjiiij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjji(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vi(index, a1) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjijjjjijjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjjjii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjijjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_fjjjj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjijjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjiiiiiijjjjjjfj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjijjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viiijjijjifjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vfjji(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiff(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjiff(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjiff(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjiiiiiijjjjjjjfji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjijjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ijjjjjjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjiifijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjffjifijjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viiiiiiffjifjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viiiiiiffjfjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viiiiiiffjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjfffjifijjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjfffifijjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjfffifijjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjfffjifjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjfffijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjifjjVj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjfjjj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_fffffff(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjfj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jfi(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjjiiiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjiiijjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjiiij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiijjjjjii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjiiijjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjijji(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjijjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjii(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_iii(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjfij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjfj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjijjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjf(index, a1, a2) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjiii(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjfjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjijj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjjjjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20, a21, a22, a23, a24, a25) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20, a21, a22, a23, a24, a25);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjiijj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjiji(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjfijfijjjij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjij(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjiiiiiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjf(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjfiijjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijijjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjiiiijjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiiiiiiii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjiijjjjjjjjii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjii(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiii(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjii(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjiijjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjijiijii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjijjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjijiijii(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjijjjiiiiiiiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjjjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19, a20);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjiiiijjij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjiiii(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjiiiiiij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjiiiiijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjiiiiiijjjjjjjjfj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17, a18, a19);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ijjjjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjf(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjjjdj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jijj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjiji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjjjjijj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjijijj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjijjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjjjij(index, a1, a2, a3, a4, a5, a6, a7, a8, a9) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjjjji(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjjij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjf(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjijij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijijjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjij(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjji(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjfi(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjijij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjij(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjijj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjij(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijjjjjjjjjjfj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_iijjj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjjijj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjjjjjijj(index, a1, a2, a3, a4, a5, a6, a7, a8) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjiij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjijjjjjjjjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jii(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjd(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjijjiii(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjf(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vjdi(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjiii(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_fjjj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijii(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjiiiji(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jij(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_djjjjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_djjj(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jjjijjj(index, a1, a2, a3, a4, a5, a6) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ijjii(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jijiii(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_vijij(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjiijjj(index, a1, a2, a3, a4, a5, a6, a7) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijijj(index, a1, a2, a3, a4) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_ijiiji(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_iijjjj(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_jiji(index, a1, a2, a3) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_jjjjid(index, a1, a2, a3, a4, a5) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
  return 0n;
 }
}

function invoke_ijjijijjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vijjjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_viijjjjjjjj(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_vjjjijjjijiijjji(index, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15) {
 var sp = stackSave();
 try {
  getWasmTableEntry(Number(index))(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_fjj(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function invoke_djj(index, a1, a2) {
 var sp = stackSave();
 try {
  return getWasmTableEntry(Number(index))(a1, a2);
 } catch (e) {
  stackRestore(sp);
  if (e !== e + 0) throw e;
  _setThrew(1, 0);
 }
}

function instrumentWasmExportsForMemory64(exports) {
 exports = Object.assign({}, exports);
 function wasm64Wrapper_p(f) {
  return function() {
   return Number(f());
  };
 }
 function wasm64Wrapper_pp(f) {
  return function(a0) {
   return Number(f(BigInt(a0)));
  };
 }
 function wasm64Wrapper__p(f) {
  return function(a0) {
   return f(BigInt(a0));
  };
 }
 function wasm64Wrapper__ppp(f) {
  return function(a0, a1, a2) {
   return f(BigInt(a0), BigInt(a1), BigInt(a2));
  };
 }
 exports["__errno_location"] = wasm64Wrapper_p(exports["__errno_location"]);
 exports["malloc"] = wasm64Wrapper_pp(exports["malloc"]);
 exports["free"] = wasm64Wrapper__p(exports["free"]);
 exports["__getTypeName"] = wasm64Wrapper_pp(exports["__getTypeName"]);
 exports["setThrew"] = wasm64Wrapper__p(exports["setThrew"]);
 exports["stackSave"] = wasm64Wrapper_p(exports["stackSave"]);
 exports["stackRestore"] = wasm64Wrapper__p(exports["stackRestore"]);
 exports["stackAlloc"] = wasm64Wrapper_pp(exports["stackAlloc"]);
 exports["__cxa_decrement_exception_refcount"] = wasm64Wrapper__p(exports["__cxa_decrement_exception_refcount"]);
 exports["__cxa_increment_exception_refcount"] = wasm64Wrapper__p(exports["__cxa_increment_exception_refcount"]);
 exports["__cxa_can_catch"] = wasm64Wrapper__ppp(exports["__cxa_can_catch"]);
 exports["__cxa_is_pointer_type"] = wasm64Wrapper__p(exports["__cxa_is_pointer_type"]);
 return exports;
}

Module["stackAlloc"] = stackAlloc;

Module["stackSave"] = stackSave;

Module["stackRestore"] = stackRestore;

Module["UTF8ToString"] = UTF8ToString;

Module["stringToUTF8"] = stringToUTF8;

Module["lengthBytesUTF8"] = lengthBytesUTF8;

var calledRun;

dependenciesFulfilled = function runCaller() {
 if (!calledRun) run();
 if (!calledRun) dependenciesFulfilled = runCaller;
};

function run() {
 if (runDependencies > 0) {
  return;
 }
 preRun();
 if (runDependencies > 0) {
  return;
 }
 function doRun() {
  if (calledRun) return;
  calledRun = true;
  Module["calledRun"] = true;
  if (ABORT) return;
  initRuntime();
  readyPromiseResolve(Module);
  if (Module["onRuntimeInitialized"]) Module["onRuntimeInitialized"]();
  postRun();
 }
 if (Module["setStatus"]) {
  Module["setStatus"]("Running...");
  setTimeout(function() {
   setTimeout(function() {
    Module["setStatus"]("");
   }, 1);
   doRun();
  }, 1);
 } else {
  doRun();
 }
}

if (Module["preInit"]) {
 if (typeof Module["preInit"] == "function") Module["preInit"] = [ Module["preInit"] ];
 while (Module["preInit"].length > 0) {
  Module["preInit"].pop()();
 }
}

run();


  return ortWasm.ready
}

);
})();
if (typeof exports === 'object' && typeof module === 'object')
  module.exports = ortWasm;
else if (typeof define === 'function' && define['amd'])
  define([], function() { return ortWasm; });
else if (typeof exports === 'object')
  exports["ortWasm"] = ortWasm;
